<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Chaya D Stern" />
  <meta name="author" content="Christopher I Bayly" />
  <meta name="author" content="Daniel G A Smith" />
  <meta name="author" content="Josh Fass" />
  <meta name="author" content="Lee-Ping Wang" />
  <meta name="author" content="David L Mobley" />
  <meta name="author" content="John D Chodera" />
  <meta name="dcterms.date" content="2020-04-20" />
  <meta name="keywords" content="forcefield, force-field, cheminformatics" />
  <title>Capturing through-bond non-local effects in torsion parameters using fractional bond orders</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Capturing through-bond non-local effects in torsion parameters using fractional bond orders" />
  <meta name="citation_title" content="Capturing through-bond non-local effects in torsion parameters using fractional bond orders" />
  <meta property="og:title" content="Capturing through-bond non-local effects in torsion parameters using fractional bond orders" />
  <meta property="twitter:title" content="Capturing through-bond non-local effects in torsion parameters using fractional bond orders" />
  <meta name="dc.date" content="2020-04-20" />
  <meta name="citation_publication_date" content="2020-04-20" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Chaya D Stern" />
  <meta name="citation_author_institution" content="Tri-Insitutional PhD Program in Chemical Biology, Memorial Sloan Kettering Cancer Center, New York, New York 10065, USA" />
  <meta name="citation_author_institution" content="Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA" />
  <meta name="citation_author_orcid" content="0000-0001-6200-3993" />
  <meta name="twitter:creator" content="@SternChaya" />
  <meta name="citation_author" content="Christopher I Bayly" />
  <meta name="citation_author_institution" content="OpenEye Scientific Software" />
  <meta name="citation_author_orcid" content="0000-0001-9145-6457" />
  <meta name="citation_author" content="Daniel G A Smith" />
  <meta name="citation_author_institution" content="The Molecular Sciences Software Institute, Blacksburg, Virginia 24060 USA" />
  <meta name="citation_author_orcid" content="0000-0001-8626-0900" />
  <meta name="twitter:creator" content="@dga_smith" />
  <meta name="citation_author" content="Josh Fass" />
  <meta name="citation_author_institution" content="Tri-Insitutional PhD Program in Chemical Biology, Memorial Sloan Kettering Cancer Center, New York, New York 10065, USA" />
  <meta name="citation_author_institution" content="Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA" />
  <meta name="citation_author_orcid" content="0000-0003-3719-266X" />
  <meta name="twitter:creator" content="@maxentile" />
  <meta name="citation_author" content="Lee-Ping Wang" />
  <meta name="citation_author_institution" content="Department of Chemistry, University of California, Davis, Califnornia 95616 USA" />
  <meta name="citation_author_orcid" content="0000-0003-3072-9946" />
  <meta name="citation_author" content="David L Mobley" />
  <meta name="citation_author_institution" content="Departments of Pharmaceutical Sciences and Chemistry, University of California, Irvine, California 92697" />
  <meta name="citation_author_orcid" content="0000-0002-1083-5533" />
  <meta name="twitter:creator" content="@davidmobley" />
  <meta name="citation_author" content="John D Chodera" />
  <meta name="citation_author_institution" content="Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA" />
  <meta name="citation_author_institution" content="BIH Einstein Visiting Professor, Charité Universitätsmedizin, Berlin" />
  <meta name="citation_author_orcid" content="0000-0003-0542-119X" />
  <meta name="twitter:creator" content="@jchodera" />
  <link rel="canonical" href="https://ChayaSt.github.io/wbo-manuscript/" />
  <meta property="og:url" content="https://ChayaSt.github.io/wbo-manuscript/" />
  <meta property="twitter:url" content="https://ChayaSt.github.io/wbo-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://ChayaSt.github.io/wbo-manuscript/" />
  <meta name="citation_pdf_url" content="https://ChayaSt.github.io/wbo-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://ChayaSt.github.io/wbo-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://ChayaSt.github.io/wbo-manuscript/v/bd38625884bd4af8b30c50534bb52da552bb47ae/" />
  <meta name="manubot_html_url_versioned" content="https://ChayaSt.github.io/wbo-manuscript/v/bd38625884bd4af8b30c50534bb52da552bb47ae/" />
  <meta name="manubot_pdf_url_versioned" content="https://ChayaSt.github.io/wbo-manuscript/v/bd38625884bd4af8b30c50534bb52da552bb47ae/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  
  <!-- pandoc-eqnos: equation style -->
  <style>
    .eqnos { display: inline-block; position: relative; width: 100%; }
    .eqnos br { display: none; }
    .eqnos-number { position: absolute; right: 0em; top: 50%; line-height: 0; }
  </style>
</head>
<body>
<header id="title-block-header">
<h1 class="title">Capturing through-bond non-local effects in torsion parameters using fractional bond orders</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://ChayaSt.github.io/wbo-manuscript/v/bd38625884bd4af8b30c50534bb52da552bb47ae/">permalink</a>)
was automatically generated
from <a href="https://github.com/ChayaSt/wbo-manuscript/tree/bd38625884bd4af8b30c50534bb52da552bb47ae">ChayaSt/wbo-manuscript@bd38625</a>
on April 20, 2020.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Chaya D Stern</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6200-3993">0000-0001-6200-3993</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/ChayaSt">ChayaSt</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/SternChaya">SternChaya</a><br>
<small>
Tri-Insitutional PhD Program in Chemical Biology, Memorial Sloan Kettering Cancer Center, New York, New York 10065, USA; Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA
· Funded by [‘NSF GRFP DGE-1257284’, ‘MolSSI NSF ACI-1547580’, ‘NSF CHE-1738979’]
</small></p></li>
<li><p><strong>Christopher I Bayly</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9145-6457">0000-0001-9145-6457</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/cbayly13">cbayly13</a><br>
<small>
OpenEye Scientific Software
</small></p></li>
<li><p><strong>Daniel G A Smith</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8626-0900">0000-0001-8626-0900</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/dgasmith">dgasmith</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/dga_smith">dga_smith</a><br>
<small>
The Molecular Sciences Software Institute, Blacksburg, Virginia 24060 USA
· Funded by [‘NSF OAC-1547580’, ‘Open Force Field Consortium’]
</small></p></li>
<li><p><strong>Josh Fass</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3719-266X">0000-0003-3719-266X</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/maxentile">maxentile</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/maxentile">maxentile</a><br>
<small>
Tri-Insitutional PhD Program in Chemical Biology, Memorial Sloan Kettering Cancer Center, New York, New York 10065, USA; Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA
· Funded by [‘NSF CHE-1738979’]
</small></p></li>
<li><p><strong>Lee-Ping Wang</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3072-9946">0000-0003-3072-9946</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/leeping">leeping</a><br>
<small>
Department of Chemistry, University of California, Davis, Califnornia 95616 USA
</small></p></li>
<li><p><strong>David L Mobley</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1083-5533">0000-0002-1083-5533</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/davidlmobley">davidlmobley</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/davidmobley">davidmobley</a><br>
<small>
Departments of Pharmaceutical Sciences and Chemistry, University of California, Irvine, California 92697
· Funded by [‘NIH R01GM132386’]
</small></p></li>
<li><p><strong>John D Chodera</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0542-119X">0000-0003-0542-119X</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/jchodera">jchodera</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/jchodera">jchodera</a><br>
<small>
Computational and Systems Biology Program, Memorial Sloan Kettering Cancer Center, New York, New York 10065 USA; BIH Einstein Visiting Professor, Charité Universitätsmedizin, Berlin
· Funded by [‘NSF CHE-1738979’, ‘NIH R01GM132386’, ‘NIH P30CA008748’, ‘Sloan Kettering Institute’]
</small></p></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Accurate small molecule force fields are crucial for predicting thermodynamic and kinetic properties of drug-like molecules in biomolecular systems.
Torsion parameters, in particular, are essential for determining conformational distribution of molecules.
However, they are usually fit to computational expensive quantum chemical torsion scans and generalize poorly to different chemical environments.
Torsion parameters should ideally capture local through-space non-bonded interactions such as 1-4 steric and electrostatics and non-local through-bond effects such as conjugation and hyperconjugation.
Non-local through-bond effects are sensitive to remote substituents and are a contributing factor to torsion parameters poor transferability.
Here we show that fractional bond orders such as the Wiberg Bond Order (WBO) are sensitive to remote substituents and correctly captures extent of conjugation and hyperconjugation.
We show that the relationship between WBO and torsion barrier heights are linear and can therefore serve as a surrogate to QC torsion barriers, and to interpolate torsion force constants.
Using this approach we can reduce the number of computational expensive QC torsion scans needed while maintaining accurate torsion parameters.
We demonstrate this approach to a set of substituted benzene rings.</p>
<h2 id="introduction">1. Introduction</h2>
<p>Molecular mechanics (MM) methods rely on Newtonian physics inspired empirical force field to describe the potential energy of the system and are widely used to study larger systems with <span class="math inline">\(10^3\)</span> - <span class="math inline">\(10^6\)</span> atoms <span class="citation" data-cites="aLuXbQN8">[<a href="#ref-aLuXbQN8" role="doc-biblioref">1</a>]</span>.
They are sufficiently cheap and accurate to study biological relevant systems and provide atomistic details of mechanisms in enzymes in solution and can reliably predict thermodynamic properties such as binding free energies [<span class="citation" data-cites="qhEZ9qxK"><a href="#ref-qhEZ9qxK" role="doc-biblioref">2</a></span>; <span class="citation" data-cites="9EY3maYh"><a href="#ref-9EY3maYh" role="doc-biblioref">3</a></span>; CITE free energy].
However, given the larger chemical space small small molecule force fields need to cover to adequately represent druglike molecules and common metabolites, they have lagged behind protein force fields <span class="citation" data-cites="mbmV3Fg8 18mvPmwwB">[<a href="#ref-mbmV3Fg8" role="doc-biblioref">4</a>,<a href="#ref-18mvPmwwB" role="doc-biblioref">5</a>]</span>.
In MM forcefields, the potential energy is constructed with terms for bond stretching, angle bending, internal rotations, electrostatics and Lennard-Jones for attractive and repulsive forces <span class="citation" data-cites="11Z8pXbEW 1GDaakPWY Mi3Ujd07">[<a href="#ref-11Z8pXbEW" role="doc-biblioref">6</a>,<a href="#ref-1GDaakPWY" role="doc-biblioref">7</a>,<a href="#ref-Mi3Ujd07" role="doc-biblioref">8</a>]</span>.
These free parameters in these functionals are fit to reproduce experimental and quantum chemical (QC) data.</p>
<h3 id="the-torsion-functional-describes-the-potential-energy-of-internal-rotation">1.1 The torsion functional describes the potential energy of internal rotation</h3>
<p>In most force fields (CHARMM <span class="citation" data-cites="11Z8pXbEW">[<a href="#ref-11Z8pXbEW" role="doc-biblioref">6</a>]</span>, AMBER <span class="citation" data-cites="1GDaakPWY">[<a href="#ref-1GDaakPWY" role="doc-biblioref">7</a>]</span>, OPLS <span class="citation" data-cites="Mi3Ujd07">[<a href="#ref-Mi3Ujd07" role="doc-biblioref">8</a>]</span>) the torsion potential is given by a truncated Fourier series:</p>
<p><span id="eq:torsion" class="eqnos"><span class="math display">\[ E_{torions} = \sum_{torsions}\sum_{n=1}^N K_{\phi, n}(1 + cos(n\phi_i - \delta_n)) \]</span><span class="eqnos-number">(1)</span></span></p>
<p>where the sum is over torsion angles <span class="math inline">\(\phi\)</span> and multiplicities <span class="math inline">\(n\)</span>.
The force constants $K_{, n} determine the barrier heights, the multiplicities <span class="math inline">\(n\)</span> determine the number of minima and the phase angles $_{, n} determines the phase.
In most force fields the <span class="math inline">\(N\)</span> can go up to 6 and the phase angles <span class="math inline">\(\delta\)</span> are usually set to <span class="math inline">\(0^{\circ}\)</span> or <span class="math inline">\(180^{\circ}\)</span> to symmetrize the curve around <span class="math inline">\(0^{\circ}\)</span> <span class="citation" data-cites="i8ZZOoVS">[<a href="#ref-i8ZZOoVS" role="doc-biblioref">9</a>]</span>.</p>
<p>The torsion energy about a bond is determined by a combination of local and non-local effects from conjugation, hypercojugation, sterics, and electrostatics <span class="citation" data-cites="fpE0Y69s AWj8hnbG Vx7ALVw2 1DVkKVgPL">[<a href="#ref-fpE0Y69s" role="doc-biblioref">10</a>,<a href="#ref-AWj8hnbG" role="doc-biblioref">11</a>,<a href="#ref-Vx7ALVw2" role="doc-biblioref">12</a>,<a href="#ref-1DVkKVgPL" role="doc-biblioref">13</a>]</span>.
Steric and elecrostatic interactions are, in principle, accounted for by non bonded terms in most force fields, so a torsion profile would ideally primarily capture conjugation or hyperconjugation effects, and only the 1-4 electrostatics.</p>
<p>The torsion functional which models internal rotation is particularly challenging to parameterize and the least transferable relative to other bonded terms for several reasons <span class="citation" data-cites="13fQUQyf9 thaStBUv fpE0Y69s">[<a href="#ref-13fQUQyf9" role="doc-biblioref">14</a>,<a href="#ref-thaStBUv" role="doc-biblioref">15</a>,<a href="#ref-fpE0Y69s" role="doc-biblioref">10</a>]</span>.
First, torsion parameters are usually fit to computationally expensive QC torsion scans which introduces a bottle neck to setting up simulations.
Second, torsion parameters are ‘soft’ or very flexible degrees of freedom compared to bond and angle parameters.
Relatively small variations to the torsional potential surface can strongly influence molecular conformation distributions <span class="citation" data-cites="thaStBUv">[<a href="#ref-thaStBUv" role="doc-biblioref">15</a>]</span>.
Given how critical torsion parameters are in determining conformation distributions in simulations, it is prudent for them to be accurately parameterized.
Third, torsion potentials can be strongly influenced by distal substituents due to changes in conjugation or hypercojugation, an effect very difficult to represent in a force field that uses only local chemistry to define parameters [CITE fragmenter paper].
When parameterizing a molecular system, the force field assigns atom types, or a description of the atom’s atomic number and chemical environment to every atom in the system <span class="citation" data-cites="jcQ6P6uD VYvjE9wM">[<a href="#ref-jcQ6P6uD" role="doc-biblioref">16</a>,<a href="#ref-VYvjE9wM" role="doc-biblioref">17</a>]</span>.
The reason atom types are used is to allow transferability of parameters to other atoms in similar chemical environments.
A torsion type is defined by the quartet of atom types of the four atoms involved in the torsion <span class="citation" data-cites="Z8BM4e7W YmRgHfeU">[<a href="#ref-Z8BM4e7W" role="doc-biblioref">18</a>,<a href="#ref-YmRgHfeU" role="doc-biblioref">19</a>]</span>.
However, these atom types are generally defined by their local chemical environment which leads to locally defined torsion types.
Therefore, non-local through-bond effects such as conjugation and hypercojugation are difficult to capture.
The inability of traditional torsion types to capture such effects and the torsion energy profile’s sensitivity distal chemical changes are contributing factors to torsion parameters’ poor transferability.</p>
<p>To address this issue of poor parameter transferability, many practitioners have employed bespoke parameterization where parameters are fit to a specific molecule.
Many automatic and semi-automatic tools exist to generate such bespoke paraemters such as GAMMP [cite], ffTK [cite], Paramfit [cite], and QUBEKit <span class="citation" data-cites="1F62Qo2RU">[<a href="#ref-1F62Qo2RU" role="doc-biblioref">20</a>]</span>.
When only a few molecules need to be parameterized, these tools are very useful for aiding researchers in setting up molecular systems in a systematic and reproducible way.
However, the paraemters generated from such tools are not meant to be generalizable and are not practical when many molecules need to be paraemterized.
Another approach to overcome this lack of transferability is to add new atom types to adequately capture torsion profiles <span class="citation" data-cites="JJX4iDz2">[<a href="#ref-JJX4iDz2" role="doc-biblioref">21</a>]</span>.
However, these new atom types are generally added in an unsystematic way and leads to a proliferation of other force field terms <span class="citation" data-cites="1HYTTY1PU">[<a href="#ref-1HYTTY1PU" role="doc-biblioref">22</a>]</span>.</p>
<p>Atom type independent force fields seek to overcome both the proliferation of force field terms and transferability by moving away from using atom types.
The SMIRNOFF force field <span class="citation" data-cites="1HYTTY1PU">[<a href="#ref-1HYTTY1PU" role="doc-biblioref">22</a>]</span> uses SMARTS patterns for direct chemical perception to replace atom types and avoid table look-ups.
In the context of torsion types, the SMIRNOFF format allows one to create a new SMARTS pattern for a torsion type without the need to introduce new atom types.
These SMARTS patterns can be created at different levels of granualarity for torsion types without worrying about proliferation of other force field terms.
The Hyperconjugation for Torsional Energy Quantification (H-TEQ) [<span class="citation" data-cites="15wrEpSer"><a href="#ref-15wrEpSer" role="doc-biblioref">23</a></span>; <span class="citation" data-cites="16GNoHAFY"><a href="#ref-16GNoHAFY" role="doc-biblioref">24</a></span>; 10.1021/acs.jcim.9b00585] approach uses
chemical principles of conjugation, hyperconjugation, and electronegativity of the atoms involved in the torsion to model torsion energies without the need to directly fit torsion parameters.
In this study we combine these two approaches by using the Wiberg Bond Order (WBO), a measure of electron population overlap between atoms in a bond, and
SMARTS torsion types to interpolate torsion force field parameters.</p>
<h3 id="fractional-bond-orders-describe-the-extent-of-bonding-between-two-atoms">1.2 Fractional bond orders describe the extent of bonding between two atoms</h3>
<p>In quantum mechanics, a molecule is a system of individual particles without explicit bonds - nuclei and electrons.
Chemists, however, think of molecules as atoms held together by bonds.
The connectivity graph of molecules is a very important mental model in chemistry, based on centuries of chemical observations and knowledge that chemists employ when thinking about molecules.
Many quantum chemists such as Pauling <span class="citation" data-cites="ZAAgsuwk">[<a href="#ref-ZAAgsuwk" role="doc-biblioref">25</a>]</span>, Coulson <span class="citation" data-cites="QY82Gb6W">[<a href="#ref-QY82Gb6W" role="doc-biblioref">26</a>]</span>, Mulliken <span class="citation" data-cites="9WKaagH4">[<a href="#ref-9WKaagH4" role="doc-biblioref">27</a>]</span>,
Wiberg <span class="citation" data-cites="BES2Ksiq">[<a href="#ref-BES2Ksiq" role="doc-biblioref">28</a>]</span>, and Mayer <span class="citation" data-cites="8uYd8cxr">[<a href="#ref-8uYd8cxr" role="doc-biblioref">29</a>]</span>, Jug <span class="citation" data-cites="vfRrdWeG">[<a href="#ref-vfRrdWeG" role="doc-biblioref">30</a>]</span>, Politzer <span class="citation" data-cites="11ndqhfsy">[<a href="#ref-11ndqhfsy" role="doc-biblioref">31</a>]</span> have worked on bridging the gap between the physical and chemical conception of atoms in molecules by an  analysis of the wave function to arrive at a fractional bond order that is consistent with the chemical concept of the multiplicity of a chemical bond.
Given that these quantities try to make that connection, it is not surprising that fractional bond orders captures important, chemical properties that can be useful in many applications, specifically force field parameterization.
Indeed, In the MMP2 <span class="citation" data-cites="mgZozbxT">[<a href="#ref-mgZozbxT" role="doc-biblioref">32</a>]</span>, MM3 <span class="citation" data-cites="F7LEPYAx">[<a href="#ref-F7LEPYAx" role="doc-biblioref">33</a>]</span> and MM4 <span class="citation" data-cites="1F0ePTGWd">[<span class="citeproc-not-found" data-reference-id="1F0ePTGWd"><strong>???</strong></span>]</span> force fields, a Variable Electronegativity SCF derived bond order for pi-systems was used to fit bond length, bond force constants and twofold torsional force constants
Here we use the Wiberg Bond Order (WBO), a relatively simple and cheap quantity that comes for free from an AM1BCC calculation used to generate partial charges to interpolate torsion force constants.
First we show that WBOs are a good indicator of the electron density around the central bond and is linearly related to torsion barrier heights.
Then we show that it is possible to use this relationship to interpolate torsion force constants for the same torsion type in different chemical environments.
Lastly, we use a data-driven approach to generate SMARTS patterns for the needed torsion types and demonstrate this approach on a set of substituted phenyl rings.</p>
<h2 id="results-and-discussion">2. Results and Discussion</h2>
<h3 id="torsion-energy-barriers-are-sensitive-to-the-chemical-environment-which-can-be-influenced-by-remote-substituents">2.1 Torsion energy barriers are sensitive to the chemical environment which can be influenced by remote substituents</h3>
<p>In most force fields, torsions are defined by the quartet of atoms types involved in the dihedral <span class="citation" data-cites="waPeRcgV 1GDaakPWY 11Z8pXbEW Mi3Ujd07">[<a href="#ref-11Z8pXbEW" role="doc-biblioref">6</a>,<a href="#ref-1GDaakPWY" role="doc-biblioref">7</a>,<a href="#ref-Mi3Ujd07" role="doc-biblioref">8</a>,<a href="#ref-waPeRcgV" role="doc-biblioref">34</a>]</span>.
Atom types encode chemical environments of atoms that usually only incorporate the local environment.
However, the quartet of atom types do not always capture all relevant chemistry, especially when the effects are nonlocal; atoms contributing to hyperconjugation, delocalization, or other nonclassical effects may not be part of the quartet involved in the torsion yet can influence the torsion profile <span class="citation" data-cites="13fQUQyf9">[<a href="#ref-13fQUQyf9" role="doc-biblioref">14</a>]</span>.
In particular, with conjugated systems, distal electron-donating or -withdrawing substituents can exert a strong effect on torsional barrier height.</p>
<p>Simple examples can help illustrate this, such as the biphenyl example in different protonation states shown in Figure <a href="#fig:biphenyls">1</a>, A.
While the MM torsion profiles are all the same (Figure <a href="#fig:biphenyls">1</a> D), the QC torsion profiles are different for each protonation state (Figure <a href="#fig:biphenyls">1</a> C).
The torsion energy barrier increases relative to the neutral state for the cation, anion, and zwitterion, in that order.
The profile changes qualitatively as well.
For the neutral molecule, the lowest energy conformer is slightly out of plane, at <span class="math inline">\(150^{\circ}\)</span> and <span class="math inline">\(120^{\circ}\)</span>.
For the others, the lowest energy conformer is at <span class="math inline">\(180^{\circ}\)</span>.
In the neutral molecule, the slightly out-of-plane conformer is preferred to accommodate the proximal hydrogens.
In the other cases, the increasing double-bond character of the conjugated central bond (shown for the zwitterion in Figure <a href="#fig:biphenyls">1</a> B) makes the planar conformer preferred.</p>
<p>This trend poses problems to generalized torsion force field parametrization.
Most general force fields consider the central bond in all tautomers equally rotatable so their MM torsion profiles are all the same (Figure <a href="#fig:biphenyls">1</a>, D), while the QC scan clearly shows that they are not.
This illustrates one of the fundamental limits of atom types in classical force fields:
At what point in this series should a new atom type be introduced?
In this case, remote changes three bonds away from the torsion central bond gradually perturbed the conjugated bond from being highly rotatable to non-rotatable as the conjugation increased.</p>
<div id="fig:biphenyls" class="fignos">
<figure>
<img src="images/figure_3_2.svg" alt="" /><figcaption><span>Figure 1:</span> <strong>Torsion profiles can be sensitive to remote substituents changes in a molecule.</strong>
<strong>[A]</strong> Biphenyl protonation states and tautomers with increasing Wiberg bond order for the central bond.
<strong>[B]</strong> The resonance structure of the biphenyl zwitterion shows how the central bond is highly conjugated. The Wiberg bond order
and torsion scan for this bond (see <strong>A</strong> and <strong>C</strong>) are reflective of a conjugated bond.
<strong>[C]</strong> Relative QC energy as a function of torsion angle of the central bond computed via QCArchive at B3LYP-D3(BJ) / DZVP level of theory.
The colors of the QC scan corresponds to the highlighted bonds in <strong>A</strong>.
<strong>[D]</strong> Same as <strong>C</strong>, but using MM energy computed via the openff-1.0.0 force field.
<strong>[E]</strong> Torsion barrier heights vs WBOs.
The color of the data points correspond to the highlighted bonds in <strong>A</strong>.
The QC torsion barrier height scales approximately linearly with the WBO.</figcaption>
</figure>
</div>
<h3 id="the-wiberg-bond-order-wbo-quantifies-the-electronic-population-overlap-between-two-atoms-and-captures-bond-conjugation">2.2 The Wiberg bond order (WBO) quantifies the electronic population overlap between two atoms and captures bond conjugation</h3>
<p>The Wiberg bond order (WBO) is a bond property that is calculated using atomic orbitals (AOs) that are used as basis sets in quantum and semi-empirical methods <span class="citation" data-cites="QlopodHU">[<a href="#ref-QlopodHU" role="doc-biblioref">35</a>]</span>.
WBOs originally started within the CNDO formalism <span class="citation" data-cites="BES2Ksiq">[<a href="#ref-BES2Ksiq" role="doc-biblioref">28</a>]</span>, but has been extended to other semi-empirical methods such as AM1 <span class="citation" data-cites="tnSqySMX">[<a href="#ref-tnSqySMX" role="doc-biblioref">36</a>]</span> and PM3 <span class="citation" data-cites="dgUvu4tX">[<a href="#ref-dgUvu4tX" role="doc-biblioref">37</a>]</span>.
The WBO is a measure of electron density between two atoms in a bond and is given by the quadratic sum of the density matrix elements over occupied atomic orbitals<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> on atoms A and B:</p>
<p><span class="math display">\[\begin{equation} W_{AB} = \sum_{\mu\in A}\sum_{\nu\in B} P^2_{\mu\nu} \end{equation}\]</span></p>
<h3 id="the-wbo-is-an-inexpensive-surrogate-for-monitoring-changes-in-the-chemical-environment-around-a-bond">2.3 The WBO is an inexpensive surrogate for monitoring changes in the chemical environment around a bond</h3>
<p>Since the WBO can be calculated from an inexpensive AM1 calculation, is indicative of a bond’s conjugation, and is correlated with torsion energy barrier height, it is attractive to use for interpolating torsion force constants.
However, WBOs are conformation-dependent <span class="citation" data-cites="r3xLNmRT hIon1SaZ">[<a href="#ref-r3xLNmRT" role="doc-biblioref">41</a>,<a href="#ref-hIon1SaZ" role="doc-biblioref">42</a>]</span>, so we further investigated this dependence to understand if WBOs will be a robust descriptor.
In addition, we also investigated the generality of the torsion energy barrier and WBO linear relationship.
In this section, we will first discuss our findings and solution to the conformation dependency and then discuss the generality of the WBO linear relationship with torsion barrier height.</p>
<h4 id="conformation-dependent-variance-of-wbos-are-higher-for-conjugated-bonds">2.3.1 Conformation-dependent variance of WBOs are higher for conjugated bonds</h4>
<p>Because they are a function of the electron density, WBOs are necessarily conformation-dependent.
However, not all bond WBOs change the same way with conformation.
We found that WBOs for conjugated bonds have higher variance with respect to conformation and that bonds involved in conjugated systems have WBOs that are correlated with each other.
This makes sense in terms of our qualitative understanding of conjugation strength depending on the alignment of <span class="math inline">\(\pi\)</span> orbitals across the conjugated system: as the change of a distal torsion disrupts the <span class="math inline">\(\pi\)</span> orbital alignment, the strength of that conjugation on the local torsion barrier decreases.</p>
<div id="fig:wbo_var_corr" class="fignos">
<figure>
<img src="images/figure-4.svg" alt="" /><figcaption><span>Figure 2:</span> <strong>Variance and correlations of Wiberg bond order distributions with respect to conformations are higher for conjugated bonds.</strong>
<strong>[A]</strong> Gefitinib, with its rotatable bonds highlighted and numbered to correspond with figures B and C.
<strong>[B]</strong> WBO distributions over 232 conformations of the highlighted, rotatable bonds.
The colors of the distributions correspond to
the colors in the highlighted bonds in A.
The variance and their 95% confidence interval are shown on the left (with exponent base of 1E-5).
The single non conjugated bonds (blue, (10, 11), red (11, 12), and olive (12, 13)) have lower variance than conjugated bonds (yellow (3, 2), purple (8, 9), orange (9, 10), grey (23, 19), and green(24, 24)).
<strong>[C]</strong> Correlation plot of WBOs every bond in Gefitinib against WBOs of all other bonds over 232 conformations.
The white lines indicate ring systems.
Bonds in conjugated systems have higher correlations between their WBOs (see the aromatic ring systems in the two lower right diagonal squares).
Both bonds (23, 19) (grey) and (24, 23) (green) have WBOs that are correlated with their neighboring ring systems, but bond (23, 19) are more correlated with the ring systems than the green bond (24, 23).
<strong>[D]</strong> Resonance structures of Gefitinib illustrate why the grey bond (23, 19) has higher variance than the green bond (24, 23) even if
both bonds are conjugated.
When the double bond is on bond (23, 19), the negative charge is on a nitrogen which is the more stable form, vs the resonance structure where the double bond is on (24, 23) with the negative charge on a carbon.
<strong>[E]</strong> The conformations of the molecule for the highest WBO and lowest in the distribution.
The mode with higher WBOs has bond (23, 19) in plane with quinazoline which allows for stronger conjugation while the mode with lower WBOs has the bond out of plane so there is less electron population overlap in out of plane conformation.</figcaption>
</figure>
</div>
<p>To investigate how WBOs change with conformation, we used Omega <span class="citation" data-cites="1G1ccz54w">[<a href="#ref-1G1ccz54w" role="doc-biblioref">43</a>]</span> to generate conformers for a set of kinase inhibitors (SI Figure <a href="#fig:kinase_inhibitors">9</a>) and calculated the WBO for each conformation from a B3LYP-D3(BJ) / DZVP <span class="citation" data-cites="17YEERBX3 flKRoCNc YPDz9hB3 CqesmIsy">[<a href="#ref-17YEERBX3" role="doc-biblioref">44</a>,<a href="#ref-flKRoCNc" role="doc-biblioref">45</a>,<a href="#ref-YPDz9hB3" role="doc-biblioref">46</a>,<a href="#ref-CqesmIsy" role="doc-biblioref">47</a>]</span> geometry optimized calculation using Psi4 <span class="citation" data-cites="YC6qkEYS">[<a href="#ref-YC6qkEYS" role="doc-biblioref">38</a>]</span>.
Omega is a knowledge-based conformer generator that uses a modified version of MMFF94s <span class="citation" data-cites="11e7CbOCj">[<span class="citeproc-not-found" data-reference-id="11e7CbOCj"><strong>???</strong></span>]</span> to score conformations.
It has been shown to accurately reproduce experimentally observed crystallography conformers in the Platinum benchmark dataset <span class="citation" data-cites="2hGWnSJL">[<a href="#ref-2hGWnSJL" role="doc-biblioref">48</a>]</span>.</p>
<p>Figure <a href="#fig:wbo_var_corr">2</a> illustrates the results for Gefitinib (Figure <a href="#fig:wbo_var_corr">2</a>, A), a representative molecule.
Figure <a href="#fig:wbo_var_corr">2</a>, B shows the distribution of WBOs for all rotatable bonds color-coded with the colors used to highlight the bonds in Gefitinib (Figure <a href="#fig:wbo_var_corr">2</a>, A).
Single carbon-carbon bonds and carbon-nitrogen bonds formed by atoms numbered 10-13 are freely rotating.
This is reflected by the tighter distribution (lower variance) of WBOs around closer to one for those bonds.</p>
<p>The bonds involving the ether oxygens and aromatic rings (formed by atoms numbered 1-3, 8-10, 19, 23-24) exhibit higher variance.
It is interesting to note the difference in the WBOs for the conjugated bonds formed by the nitrogen between the quinazoline and chloro fluoro phenyl (bonds formed by atoms numbered 19, 23 and 23, 24).
Both of these bonds are conjugated with their neighboring ring systems.
However, while the distribution of WBOs for bond 23-19 (the grey distribution) has two clear modes of almost equal weights, the WBO distribution for bond 24-23 has lower variance.
This is in agreement with the resonance structures shown in Figure <a href="#fig:wbo_var_corr">2</a>.</p>
<p>The resonance structures that have the double bond on the bond closer to the quinazoline (bond 19-23) are more stable because the negative charge is on a nitrogen.
When the double bond is on the neighboring 23-24 bond, the negative charge is on an aromatic carbon which is less stable.
The results are similar for other kinase inhibitors tested shown in SI Figure <a href="#fig:kinase_inhibitors">9</a>.</p>
<p>In addition, when we inspected the conformations associated with the highest and lowest WBO in the grey distribution (Figure <a href="#fig:wbo_var_corr">2</a>, E) we found that conformations with lowest WBO on bond 19-23 had that bond out of plane while the conformation with the highest WBO value had the bond in plane which allows conjugation.
We found similar results from WBOs calculated form QC torsion scans.
Figure <a href="#fig:sub_phenyl_qc_scan">5</a> shows the WBO for each point in the QC corresponding torsion scans.
The WBOs are anti-correlated with the torsion potential energy which is in line with chemical intuition.
Conjugation stabilizes conformations and leads to more electronic population overlap in bonds <span class="citation" data-cites="14qKIxxbD">[<a href="#ref-14qKIxxbD" role="doc-biblioref">49</a>]</span>.
At higher energy conformers, the aromatic rings are out of plane and cannot conjugate.
Therefore the WBO is lower for those conformers.
At lower energy conformations, the rings are in plane and can conjugate so the WBO is higher.
We found that the trends discusses above are similar when using semi-empirical methods such as AM1 (SI Figure <a href="#fig:si_wbo_dist">10</a>).
For other levels of QC theory and for the Mayer bond order, the results can be different as discussed in the SI.</p>
<h4 id="bonds-in-conjugated-systems-have-highly-correlated-conformation-dependent-wbos">2.3.2 Bonds in conjugated systems have highly correlated conformation-dependent WBOs</h4>
<p>We found that certain bond orders are strongly correlated or anti-correlated with each other, indicating strong electronic coupling.
As bonds in one conformation gain electron population overlap, the coupled bonds will decrease in electron population overlap and vice versa.
Figure <a href="#fig:wbo_var_corr">2</a>, C shows the Pearson correlation coefficient for each bond WBO distribution against all other bond WBO distributions.
There is a clear structure in this correlation plot. The square formed by bonds from atoms 24-29 shows that the alternating bonds in the aromatic ring (25-29) are strongly anti-correlated with each other.</p>
<p>This trend is different in the ring formed by atoms 13-18, which is not aromatic.
In this ring, bonds 13-18, 13-14, 16-15 and 16-17 (which involve electron rich atoms O and N) have Pearson correlation coefficients with absolute values higher than for the other bonds in the ring, but lower than the bonds in the aromatic ring.
The bonds involved in the methoxy groups (atoms 1-3 and 8-10) are correlated with each other and also correlated to the quinazoline, albeit not as strongly.
The bonds between the chloro fluoro phenyl and quinazoline follow the same trend as their WBO distribution and resonance structures.
The bond closer to the quinazoline (bond 23-19) has WBO distribution correlated with the quinazoline while the bond closer to the
chloro fluoro phenyl (bond 23-24) is not as strongly coupled with the quinazoline.</p>
<p>The trends are similar for other kinase inhibitors examined, as shown in SI Figure <a href="#fig:kinase_inhibitors">9</a>.</p>
<h4 id="electronically-least-interacting-functional-group-elf-method-provides-a-useful-way-to-capture-informative-conformation-independent-wbos">2.3.3 Electronically least-interacting functional group (ELF) method provides a useful way to capture informative conformation-independent WBOs</h4>
<p><code class="yellow">[Note: Change these distributions to be over subsituted phenyl set?]</code>.
<img src="images/standard_deviations.svg" id="fig:standard_deviations" alt="Standard deviations of conformation-dependent WBO distributions are smaller than standard deviations of WBO distribution of the same bond in different chemical environments. The distribution of standard deviations of WBO distributions over conformations is shown in blue. The distribution of standard deviations of ELF10 WBO distributions for the same bond in different chemical environments is shown in pink. The changes in WBO due to conformations are smaller than the changes in WBO due to chemical changes around the bond. The validation set (SI Figure ??), was used to generate these distributions. The blue standard deviation distribution was calculated over 140602 WBO dependent distributions (this is the number of individual fragments in the dataset). The pink standard deviation distribution was calculated over 366 distributions of 366 bonds in different chemical environments (of the 140602 fragments)" /></p>
<p>As we have shown, the WBO is conformation-dependent and this dependency can also be highly informative of the electronic couplings within a system.
Figure ?? shows the distribution of standard deviations of the conformation-dependent WBO distribution in blue.
Most of the standard deviations fall below 0.03, which is encouragingly small.
However, it can become computationally expensive to calculate the WBO for all conformations; if we aim to use WBOs as a descriptor for torsion barrier heights in a reproducible way, we need a way to capture informative conformation-independent WBOs.
The Electronically Least-interacting Functional groups (ELF) conformation selection scheme implemented in the OpenEye Toolkit <code>quacpac</code> module <span class="citation" data-cites="16hZpgNca">[<a href="#ref-16hZpgNca" role="doc-biblioref">50</a>]</span> resolves the issue of sensitivity of molecular mechanics electrostatic energies from QC derived charges.</p>
<p>The ELF10 method begins with a large set of conformers for the molecule.
MMFF94 charges <span class="citation" data-cites="11e7CbOCj">[<span class="citeproc-not-found" data-reference-id="11e7CbOCj"><strong>???</strong></span>]</span> are assigned to the molecule, set to their absolute value, and then single-point Coulomb electrostatic energies evaluated for each conformer.
The lowest-energy 2% of conformers are selected, and if there are more than 10, from these the most diverse 10 are selected.
For this final conformer set (up to 10 conformers), the AM1 WBOs and charges for each conformer are averaged (by bond and by atom, respectively) and the BCCs are applied to the charges <span class="citation" data-cites="LxrgIkt0">[<a href="#ref-LxrgIkt0" role="doc-biblioref">51</a>]</span>.
This method yields a set of AM1-BCC atomic partial charges and WBOs for the molecule which are relatively insensitive to the initial choice of conformer set, and which mitigate two pathologies of AM1-BCC charges: peculiar charges resulting from strong intramolecular electrostatic interactions (e.g. from internal hydrogen bonds or formal charges) and simple conformational variance.</p>
<div id="fig:wbo_dists" class="fignos">
<figure>
<img src="images/figure-2_4.svg" alt="" /><figcaption><span>Figure 3:</span> <strong>Distribution of WBO in drug-like molecules is concentrated near chemically sensible values.</strong>
<strong>[A]</strong> The distribution of all WBOs for all bonds in molecules in set. The mode at one, close to two and close to three correspond to
single, double and triple bonds. the density between one and two correspond to aromatic and conjugated bonds. The mode at ~0.7 correspond
to bonds that include sulfur and phosphorous which are longer, weaker bonds. <strong>[B]</strong> The blue distribution includes carbon - carbon bonds that
are not in rings. The modes at one, two and three correspond to single, double and triple bonds. The pink distribution include bonds that are in rings.
The mode at one corresponds to single bonds and the density between one and 1.5 are aromatics. <strong>[C]</strong> The blue distribution includes bonds that have
either one or two nitrogens. Many of these bonds are conjugated as demonstrated by the density around 1.5. The density at three corresponds to
nitriles. The pink distribution include bonds that have oxygens. The mode at two corresponds to carbonyls.</figcaption>
</figure>
</div>
<p>This method can also be applied to produce WBOs that are insensitive to conformers.
To check how well AM1 ELF10 estimated WBOs recapitulates the multiplicity of bonds, we calculated WBOs from AM1 ELF10 calculations for all bonds in a set of molecules shown in SI Figure <a href="#fig:wbo_dist_mols">11</a>.
The distribution in Figure <a href="#fig:wbo_dists">3</a> corresponds closely with bond multiplicity.
The density at ~0.7 correspond to bonds involving sulfur and phosphorous since these are weaker and longer bonds.
The mode at ~1.0 corresponds to C-H and C-C bonds, the mode close to 1.5 corresponds to bonds in aromatic rings, the mode close to 2.0 corresponds to double bonds, and finally the triple bonds form the last peak.</p>
<p>Figures <a href="#fig:wbo_dists">3</a> B and D separate out different kinds of bonds to more clearly illustrate what the WBO captures.
Figure <a href="#fig:wbo_dists">3</a> B shows carbon - carbon bonds not in rings (blue) and bonds in rings (pink).
The carbon-carbon distribution has distinct modes at one, two and three corresponding to single, double and triple bonds. There is also a smaller mode at 1.5 that corresponds to conjugated bonds.
The pink distribution includes bonds in rings and has modes at one and 1.5 which corresponds to aliphatic and aromatic rings, respectively.
Figure <a href="#fig:wbo_dists">3</a> D shows distributions with bonds that have nitrogens (blue) and oxygens (pink).
The peaks occur at chemically sensible values; 1, 1.5 and 3 for nitrogen which corresponds to single, conjugated and triple bonds and 1 and 2 for oxygens which correspond to single and carbonyl bonds.
For the rest of this section, we focus on the robustness and generalizability of ELF10 WBOs.</p>
<h4 id="wbos-are-a-robust-signal-of-how-torsion-barrier-heights-depend-on-remote-chemical-changes">2.3.4 WBOs are a robust signal of how torsion barrier heights depend on remote chemical changes</h4>
<p>To investigate how resonance and electronic effects from remote substituents change the torsion energy of a bond, we took inspiration from the Hammett equation <span class="citation" data-cites="3z27USyZ">[<a href="#ref-3z27USyZ" role="doc-biblioref">52</a>]</span> of reactions involving benzoic acid derivatives.
The Hammett equation relates meta and para benzoic acid substituents to the acid’s ionization equilibrium constants</p>
<p><span class="math display">\[\begin{equation} log \frac{K}{K_0} = \sigma\rho \end{equation}\]</span></p>
<p>Here, <span class="math inline">\(\sigma\)</span> is a substituent constant and <span class="math inline">\(\rho\)</span> is a reaction constant.
It aims to isolate the resonance and inductive effects
of substituents from the sterics effects of a reaction. Here, we generated a combinatorial set of meta- and para-substituted phenyls
and pyridine (Figure <a href="#fig:substituted_phenyls">4</a>, A) with 26 functional groups that cover a wide range of electron donating and withdrawing groups.
We then calculated the AM1 ELF10 WBO for the bond attaching the functional group to the aromatic ring (highlighted green in
Figure <a href="#fig:substituted_phenyls">4</a>, A) for all functional groups which resulted in 133 (26 * 5 + 3) WBOs for each functional group in different chemical environments.
This allowed us to isolate the effect on a bond’s WBO from remote chemical environment changes, defined as a change more than two bonds away, from other effects such as sterics and conformations.
The resulting distributions are in Figure <a href="#fig:substituted_phenyls">4</a>, B.
(Details on generating and accessing this set are provided in Section <code class="red">[hold]</code> in the Detailed Methods.)</p>
<p>It is interesting to note that the trend of decreasing WBOs for more electron donating groups are anticorrelates with increasing Hammett substituent constants.
In SI Figure <a href="#fig:hammett">12</a>, the AM1 ELF10 WBOs of the bonds between the functional group and benzoic acid are plotted against their Hammett meta and para substituent constants (values were taken from Hansch et al. <span class="citation" data-cites="1EJC88xIC">[<a href="#ref-1EJC88xIC" role="doc-biblioref">53</a>]</span>).
Functional groups that are more electron donating will have more electron density on the bond attaching the functional group to the benzoic acid.
The resonance and/or inductive effect destabilize the benzoate, increases its pKa, which corresponds to lower substituent constants.</p>
<div id="fig:substituted_phenyls" class="fignos">
<figure>
<img src="images/substituted_phenyls.svg" alt="" /><figcaption><span>Figure 4:</span> <strong>AM1 ELF10 Wiberg bond orders correlate with torsion barrier heights in related molecules.</strong>
<strong>[A]</strong> Systems and functional groups used in the substituted phenyl set. The functional groups were chosen to span a large range of electron donating and withdrawing groups.
<strong>[B]</strong> AM1 ELF10 WBO distributions for the bond between the phenyl ring and X<sub>1</sub> in different chemical environments.
<strong>[C]</strong> Selected QC torsion scan barrier heights vs AM1 ELF10 WBOs. These lines all had r<sup>2</sup> greater than 0.7.
<strong>[D]</strong> Same as <strong>C</strong> but these series did not fit the lines as well (r<sup>2</sup> less than 0.7).</figcaption>
</figure>
</div>
<div id="tbl:stats" class="tablenos">
<table>
<caption><span>Table 1:</span> <strong>Slope and associated statistics for torsion barrier height vs WBO for selected functional groups.</strong> </caption>
<thead>
<tr class="header">
<th>X<sub>1</sub></th>
<th>slope</th>
<th>standard error</th>
<th>r<sup>2</sup> and 95% CI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>N(Me)<sub>2</sub></td>
<td>116.92</td>
<td>14.35</td>
<td><span class="math inline">\(\textsf{0.88}_{\textsf{0.73}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="even">
<td>NHMe</td>
<td>134.52</td>
<td>16.19</td>
<td><span class="math inline">\(\textsf{0.90}_{\textsf{0.83}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="odd">
<td>NH<sub>2</sub></td>
<td>64.27</td>
<td>20.76</td>
<td><span class="math inline">\(\textsf{0.58}_{\textsf{0.06}}^{\textsf{0.95}}\)</span></td>
</tr>
<tr class="even">
<td>NHEt</td>
<td>119.51</td>
<td>19.98</td>
<td><span class="math inline">\(\textsf{0.84}_{\textsf{0.61}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="odd">
<td>NH(C<sub>3</sub>H<sub>7</sub>)</td>
<td>163.76</td>
<td>23.81</td>
<td><span class="math inline">\(\textsf{0.87}_{\textsf{0.73}}^{\textsf{0.99}}\)</span></td>
</tr>
<tr class="even">
<td>OH</td>
<td>154.82</td>
<td>35.67</td>
<td><span class="math inline">\(\textsf{0.73}_{\textsf{0.26}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="odd">
<td>OMe</td>
<td>185.31</td>
<td>41.33</td>
<td><span class="math inline">\(\textsf{0.80}_{\textsf{0.46}}^{\textsf{0.99}}\)</span></td>
</tr>
<tr class="even">
<td>OEt</td>
<td>119.66</td>
<td>47.12</td>
<td><span class="math inline">\(\textsf{0.48}_{\textsf{0.07}}^{\textsf{0.88}}\)</span></td>
</tr>
<tr class="odd">
<td>NHCON(Me)<sub>2</sub></td>
<td>159.31</td>
<td>47.98</td>
<td><span class="math inline">\(\textsf{0.58}_{\textsf{0.23}}^{\textsf{0.95}}\)</span></td>
</tr>
<tr class="even">
<td>NHCONHMe</td>
<td>127.65</td>
<td>55.03</td>
<td><span class="math inline">\(\textsf{0.43}_{\textsf{0.04}}^{\textsf{0.95}}\)</span></td>
</tr>
<tr class="odd">
<td>NHCONH<sub>2</sub></td>
<td>238.12</td>
<td>54.12</td>
<td><span class="math inline">\(\textsf{0.73}_{\textsf{0.41}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="even">
<td>NHCOEt</td>
<td>205.80</td>
<td>51.80</td>
<td><span class="math inline">\(\textsf{0.69}_{\textsf{0.31}}^{\textsf{0.99}}\)</span></td>
</tr>
<tr class="odd">
<td>NHCOMe</td>
<td>144.32</td>
<td>64.12</td>
<td><span class="math inline">\(\textsf{0.46}_{\textsf{0.02}}^{\textsf{0.99}}\)</span></td>
</tr>
<tr class="even">
<td>OCONH<sub>2</sub></td>
<td>172.72</td>
<td>84.85</td>
<td><span class="math inline">\(\textsf{0.51}_{\textsf{0.03}}^{\textsf{0.98}}\)</span></td>
</tr>
<tr class="odd">
<td>COOH</td>
<td>267.23</td>
<td>91.46</td>
<td><span class="math inline">\(\textsf{0.74}_{\textsf{0.04}}^{\textsf{1.0}}\)</span></td>
</tr>
<tr class="even">
<td>COOEt</td>
<td>149.01</td>
<td>63.23</td>
<td><span class="math inline">\(\textsf{0.58}_{\textsf{0.13}}^{\textsf{1.0}}\)</span></td>
</tr>
<tr class="odd">
<td>NO<sub>2</sub></td>
<td>302.07</td>
<td>47.74</td>
<td><span class="math inline">\(\textsf{0.91}_{\textsf{0.79}}^{\textsf{1.0}}\)</span></td>
</tr>
</tbody>
</table>
</div>
<p>To investigate how these long range effects observed in the WBOs capture changes in the bonds’ torsion potential energy, we ran representative QC torsion scans for 17 of these functional groups (SI Figures <a href="#fig:phenyl_td_1">13</a>, <a href="#fig:phenyl_td_2">14</a>, <a href="#fig:phenyl_td_3">15</a>, <a href="#fig:phenyl_td_4">16</a> <a href="#fig:phenyl_td_5">17</a>).
We did not run QC torsion scans for the following functional groups:
* functional groups did not have a torsion such as halogens
* functional groups that were congested (such as trimethyl amonium)
* functional groups where the WBOs did not change by more than 0.01 for different functional groups at the meta or para position such as methyl.
We chose the representative molecules for the 17 functional groups by sorting the molecules within each functional group by their WBO and selecting molecules with minimum WBO difference of 0.02.
All of the resulting QC torsion scans are shown in SI Figures <a href="#fig:phenyl_td_1">13</a>, <a href="#fig:phenyl_td_2">14</a>, <a href="#fig:phenyl_td_3">15</a>, <a href="#fig:phenyl_td_4">16</a>, <a href="#fig:phenyl_td_5">17</a>.
Table <a href="#tbl:stats">1</a> lists the slopes and associated statistics for the fitted lines.</p>
<p>When we compare the standard deviations of WBO distributions with respect to conformation versus with respect to changes in chemical space (Figure ??, pink distribution), we find that the changes in ELF10 WBO for remote chemical environment changes are usually bigger than the changes in WBO that arise from change in conformation.
This allows us to use ELF10 WBOs as a good surrogate to changes the chemical environment that is reflected in QC torsion scans.</p>
<h4 id="conformation-dependent-wbos-differentiate-between-though-space-and-through-bond-non-local-effects-on-qc-torsion-scans">2.3.5 Conformation-dependent WBOs differentiate between though-space and through-bond non-local effects on QC torsion scans</h4>
<p>We show representative series of torsion scans in Figure <a href="#fig:sub_phenyl_qc_scan">5</a> and the corresponding conformation-dependant WBO at each conformation in the scan.
A QC torsion scan contains contributions of through-bond effects such as conjugation and / or hyperconjugation and through-space effects such as sterics and electrostatics interactions [<span class="citation" data-cites="fpE0Y69s"><a href="#ref-fpE0Y69s" role="doc-biblioref">10</a></span>; 10.1021/jp046290d].
In this section we show how WBOs can be used to characterize through-space and through-space effects in QC torsion scans.</p>
<p>As a molecule is rotated about its bond to generate a QC torsion scan, changes in conjugation and / or hyperconjugation, the conformation of the rest of the molecule, and non-local, through-space interactions contribute to the potential energy surface that is then used to fit MM torsion parameters.
The torsion parameters in classical force fields are supposed to include both conjugation, a through-bond electron delocalization phenomenon that is not well modeled in classical forcefields, and corrections for 1-4 non-bonded interactions.
To increase their transferability, torsion parameters should not include non-bonded interactions beyond the 1-4 atoms.
However, in general, it is difficult to separate the contributions of sterics and conjugation in a QC torsion scan.
In this section we characterize steric and conjugation and / or hyperconjugation contributions to QC scans using the corresponding WBO scans.
Below is a summary of these observations from the subsituted phenyl torsion scans.</p>
<ol type="1">
<li>When QC torsion scans are anti-correlated with conformation-dependent WBOs calculated for conformations in the scan, differences in QC torsion scans for the same torsion types in different chemical environments are a result of non-local, through-bond effects.</li>
<li>If changes in QC torsion scans are not accompanied by highly anti-correlated WBOs, the changes are due to non-local, through-space, effects</li>
<li>When AM1 ELF10 WBOs do not obey the trend found in Figure <a href="#fig:substituted_phenyls">4</a> of the linear relationship with torsion barrier height for the same torsion type in different chemical environments, they are generally caused by non-local, through-space interactions.</li>
</ol>
<div id="fig:sub_phenyl_qc_scan" class="fignos">
<figure>
<img src="images/example-scans.svg" alt="" /><figcaption><span>Figure 5:</span> <strong>Wiberg bond orders are anticorrelated with QC torsion scans.</strong>
<strong>[A]</strong> Selected series of molecules with central torsion bonds connecting the nitro group to the phenyl ring highlighted and labeled with AM1 ELF10 WBOs.
<strong>[B]</strong> QC torsion scans for nitro series in different chemical environments shown in <strong>A</strong>.
The color of the scans correspond to the colors of the highlighted bonds in <strong>A</strong>.
Torsion energy barriers increase with increasing AM1 ELF10 WBOs.
<strong>[C]</strong> Wiberg bond orders calculated at each point in the QC torsion scan using the same level of theory.
<strong>[D]</strong> Conformer energy of torsion scan plotted against it WBO. All molecules in
this series have WBO that are anti-correlated with their QC torsion scan. Pearson correlation coefficients (<span class="math inline">\(\rho\)</span>) are shown in the upper right legend.
<strong>[E]</strong> Same as <strong>A</strong> but with urea at the X<sub>1</sub> position.
<strong>[F]</strong> Same as <strong>B</strong> but for urea in a series of different chemical environment. Both profiles and
energy barriers change with AM1 ELF10 WBOs. In addition, the grey scan has higher energy barriers than the olive scan but its ELF10 WBO is lower.
<strong>[G]</strong> Same as <strong>C</strong> but for urea. Here, the WBO scans all have the same profiles while the QC torsion scan does not.
<strong>[H]</strong> Same as <strong>D</strong> but here the WBO scans are not always anti-correlated or not as strongly anti-correlated.
The WBO scan profiles do not change because the changes in the QC torsion scan captures spatial effects while the WBO scans capture conjugation.
<strong>[I]</strong> Same as <strong>A</strong> with amino at the X<sub>1</sub> position.
<strong>[J]</strong> Same as <strong>B</strong> but for amino in different chemical environments. While the gold scan is symmetric around 0<sup><span class="math inline">\(\circ\)</span></sup>, the red
and blue scan are not. The Blue scan is shifted and the red scan has a higher barrier on one side.
<strong>[K]</strong> Same as <strong>C</strong> but for amino. Here the WBO scans are anticorrelated with the QC torsion scans even if the QC scans have different profiles.
<strong>[L]</strong> Same as <strong>D</strong> but for amino. Here the WBO scans are atnicorrelted with the QC torsion scans.</figcaption>
</figure>
</div>
<p>Figure <a href="#fig:sub_phenyl_qc_scan">5</a> shows three series of QC torsion scans for the same torsion type in different chemical environments to illustrate this.
Figure <a href="#fig:sub_phenyl_qc_scan">5</a> A, E and I show three torsion types (nitro, urea and amino) in different chemical environments with their associated AM1 ELF10 WBO.
Figure <a href="#fig:sub_phenyl_qc_scan">5</a> B, F, and J show their QC torsion scans, <a href="#fig:sub_phenyl_qc_scan">5</a> C, G, and K show their conformation-dependent WBOs along the torsion scan, and <a href="#fig:sub_phenyl_qc_scan">5</a> D, H, and L show the correlations between conformer energies and their conformer-dependent WBOs.</p>
<p>Figure <a href="#fig:sub_phenyl_qc_scan">5</a> A-D show what is generally the expected behavior of QC torsion scans for the same torsion types in different chemical environments as shown in <a href="#fig:substituted_phenyls">4</a>.
The QC torsion profiles in ??, B, are all the same, while the torsion barrier heights increase with increasing AM1 ELF10 WBOs.
WBOs calculated at every point of the QC torsion scan generate a profile that is generally anti-correlated to QC torsion scans as shown in <a href="#fig:sub_phenyl_qc_scan">5</a>, C and D.
This is in line with chemical intuition.
Increased conjugation is a result of increased electron population overlap which stabilizes conformations and decreases their energy.</p>
<p>The second derivative of WBOs along a QC torsion scan also changes depending on how strongly the bond conjugates at its lowest energy conformation.
For bonds with higher AM1 ELF10 WBOs (Figure <a href="#fig:sub_phenyl_qc_scan">5</a>, A and E) which indicates increased conjugation, the rate of change in the WBO scans are higher than for the same torsion types in environments where the bond does not conjugate as strongly (Figure <a href="#fig:sub_phenyl_qc_scan">5</a>, C and G).
This also makes sense with respect to chemical intuition.
At high energy conformations, where electronic orbitals are not oriented to conjugate, WBOs of the same bond in different chemical environments will be closer to each other than at lower energy where electronic delocalization is also a function of long distance chemical changes.
In other words, conjugation is disrupted similarly for the different chemical environment, but the extent of conjugation is different because of different remote substituents.</p>
<p>Figure <a href="#fig:sub_phenyl_qc_scan">5</a> E-H shows a different series of the same torsion type in different chemical environments.
There are several differences between this example and the example discussed in the previous paragraph.
One, while the torsion types are equivalent for all four molecules shown in Figure <a href="#fig:sub_phenyl_qc_scan">5</a> E, the QC torsion scans do not all have the same profile.
Relative heights of minima and maxima are different or new minima and maxima are observed.
In addition, their corresponding conformation-dependent WBO scans are not as strongly anti-correlated with the torsion scans (Figure <a href="#fig:sub_phenyl_qc_scan">5</a>, H) as in Figure <a href="#fig:sub_phenyl_qc_scan">5</a> D.
Interestingly, the WBO scans do all have similar profiles Figure <a href="#fig:sub_phenyl_qc_scan">5</a> G).
In this example, the urea is a more bulky functional group than the nitro and amino group in Figures <a href="#fig:sub_phenyl_qc_scan">5</a> A and I.
The red and grey molecules also have bulky groups at the meta position relative to the torsion being driven.
This creates different steric clashes for the different molecules and changes their QC torsion profiles relative to each others.
However, the conformation-dependant WBO scans have similar profiles for the molecules in the series because the non-local through-bond effects are similar for the molecules in this series.
These WBO scans are not as anti-correlated to the QC scan as in the previous examples (-0.98, -0.99, -1.00 vs 0.7, -0.61, -0.97 and -0.67).
The blue scan is actually correlated instead of being anti-correlated.
Because while the electrons can conjugate when the torsion is in a planar position, the clashes of the proximal hydrogen and oxygen increase the energy.
This small barrier at 0<span class="math inline">\(^\circ\)</span> does not exist in the grey and gold scan because the trimethylamonium is electron withdrawing, that the urea group can better conjugate with the phenyl ring and this stabilization is greater than the steric interactions.</p>
<p>This series also shows an example of molecules that do not follow the trend shown in Figure <a href="#fig:substituted_phenyls">4</a> where increasing AM1 ELF10 WBO corresponds to increasing torsion barrier heights.
The grey torsion scan (Figure <a href="#fig:sub_phenyl_qc_scan">5</a> F) has torsion barrier heights that is ~15 kJ/mol greater than the gold torsion scan, while its AM1 ELF10 WBO is lower than the gold molecules (1.10 vs 1.11).
This happens because the trimethylamonium is a bulky group at the meta position and it interacts with the carbonyl in the urea which causes the the barrier heights to increase.
We observed this trend of bulky groups on the meta position clashing in most other cases where the AM1 ELF10 WBOs did not follow the trend of increasing torsion barrier heights (SI Figures <a href="#fig:phenyl_td_1">13</a>, <a href="#fig:phenyl_td_2">14</a>, <a href="#fig:phenyl_td_3">15</a>, <a href="#fig:phenyl_td_4">16</a>, <a href="#fig:phenyl_td_5">17</a>).</p>
<div id="fig:improper" class="fignos">
<figure>
<img src="images/example-improper-angles.svg" alt="" /><figcaption><span>Figure 6:</span> <strong>Improper torsion angles can be coupled with torsion angles being driven in torsion scans.</strong>
Improper angles of pyramidal nitrogen involved in the torsion scan in Figure <a href="#fig:sub_phenyl_qc_scan">5</a>, J.
In the blue scan, the trivalent nitrogen is in a pyramidal conformation for the entire scan and the pyramid does not inter-convert.
In the red scan, the trivalent nitrogen is also in a pyramidal conformation for the entire scan, but the pyramid does inter-convert.
In the gold scan, the pyramidal nitrogen becomes planar at torsion angles 0<span class="math inline">\(^\circ\)</span> and 180<span class="math inline">\(^\circ\)</span> as the amino group conjugates with the phenyl.</figcaption>
</figure>
</div>
<p>Figures <a href="#fig:sub_phenyl_qc_scan">5</a> I-L show yet another series of torsion scans that exhibit different behaviors than both examples already discussed.
In this example, QC torsion scans in <a href="#fig:sub_phenyl_qc_scan">5</a> J all have different profiles, specifically, the symmetry around zero is lost in the blue and red scan, albeit differently.
The blue scan shifted such that the minima are not at 0<span class="math inline">\(^\circ\)</span> and 180<span class="math inline">\(^\circ\)</span> but the barrier heights are equivalent.
In the red scan, one barrier is ~10 kJ/mol higher than the other.
However, in these scans, the corresponding WBO scans are anti-correlated with the torsion scans (Figure <a href="#fig:sub_phenyl_qc_scan">5</a>, L).
Because in these cases, these changes are due to changes in conjugation, not because of non-local steric interactions.
All of these molecules contain a trivalent nitrogen that can assume both a pyramidal and planar conformation, depending on the amount of electron density in the lone pair.
The more electron density there is on the lone pair, the greater the angle of the pyramidal nitrogen will be.
If the lone pairs conjugate with other <span class="math inline">\(\pi\)</span> electrons, the trivalent nitrogen adopts a planar conformation to accommodate conformation and because there is less electron density in the lone pair now that it is delocalized.
Figure <a href="#fig:improper">6</a> shows how the angle of the trivalent nitrogen involved in the torsion changes over the course of the torsion scan.
The blue scan, which corresponds with the molecule with a low AM1 ELF10 WBO relative to other bonds of the same torsion type (1.06), does not conjugate with the phenyl ring and remains in a pyramidal with an angle of ~ 35<span class="math inline">\(^\circ\)</span>.
This creates a chiral center which is reflected in the shift of the blue torsion scan in Figure <a href="#fig:sub_phenyl_qc_scan">5</a> J and the loss of symmetry around zero.
The red scan which also corresponds to a molecule with a lower AM1 ELF10 WBO relative to other bonds with this torsion type (1.10), does not conjugated and remains pyramidal throughout the entire scan.
However, in this scan, the chirality of the pyramidal nitrogen does flip, but then does not flip again at ~100<span class="math inline">\(^\circ\)</span> which can explain why the barrier heights are so different from one another.
Lastly, the gold scan which corresponds to a molecule with a relatively high AM1 ELF10 WBO (1.24) does conjugate at a torsion angle of zero and the trivalent nitrogen becomes planar with an improper torsion angle of zero.</p>
<p>The general trend we found when looking at other molecules in the subsituted phenyl set (SI Figures <a href="#fig:phenyl_td_1">13</a>, <a href="#fig:phenyl_td_2">14</a>, <a href="#fig:phenyl_td_3">15</a>, <a href="#fig:phenyl_td_4">16</a>, <a href="#fig:phenyl_td_5">17</a>) was that when trivalent nitrogen were involved in a torsion scan, changes in the torsion profile relative to the same torsion type in different chemical environments, were due to changes in non-local through-bond effects.
These changes were also universally reflected in the corresponding WBO scans that remained anti-correlated with the QC scan.
Specifically, torsions involving trivalent nitrogens with relatively lower AM1 ELF10 WBOs for the central bond, were more likely to exhibited such changes.
In all cases where such changes were observed, the trivalent nitrogens did not form planar conformations at appropriate points in the scan to conjugate with the neighboring phenyl ring.</p>
<p>Calculating conformation-dependent WBOs for torsion scan is computationally cheap relative to a full QC scan, and the information gleaned from it can be helpful in differentiating non-local through-bond and through-space effects.
In general, when the conformation-dependent WBO scan is not strongly anti-correlated with the QC torsion scan, the QC scan contains through-space steric effects.
When the conformation-dependent WBO scans are strongly anti-correlated with QC scans, especially when those profiles have loss of symmetry and the torsion atoms include a trivalent nitrogen, the changes in QC torsion profiles relative to QC scans of the same torsion type in different chemical environments are usually a result from through-bond effects and need to be incorporated in classical torsion force field parameters.</p>
<h3 id="wiberg-bond-orders-can-be-used-to-interpolate-torsion-parameters">2.4 Wiberg bond orders can be used to interpolate torsion parameters</h3>
<p>Outline for this section</p>
<ol type="1">
<li>Find the difference potential of the substituted phenyl torsion scans between MM and QM without torison paramters</li>
<li>Cluster on these profiles and create torsion SMARTS for them</li>
<li>Interpolate torsion parameters using ELF10 WBOs</li>
</ol>
<p>Figures to add:</p>
<ol type="1">
<li>Difference potentials clustered for similar profiles</li>
<li>Table of SMARTS patterns for the different torsion types</li>
<li>Figure showing MM torsion scans generated with interpolated torsion parameters.</li>
</ol>
<h2 id="introduction-1">3. Introduction</h2>
<h2 id="detailed-method">5. Detailed method</h2>
<h3 id="qcarchive-data-generation-and-archiving">5.1 QCArchive data generation and archiving</h3>
<p>The MolSSI QCArchive <span class="citation" data-cites="QS8UBx4">[<a href="#ref-QS8UBx4" role="doc-biblioref">54</a>]</span> project is a platform for computing, organizing, and sharing quantum chemistry data.
Computations with this platform automatically record all input and output quantities ensuring the reproducibility of all computations involved.
In the case of computing with the MolSSI QCArchive instances, all data is automatically hosted and can be queried using the platform.</p>
<h4 id="submitting-computations-to-qcarchive">5.1.1 Submitting computations to QCArchive</h4>
<p>All scripts used to submit calculations for the datasets in this paper are in the <a href="https://github.com/openforcefield/qca-dataset-submission"><code>qca-dataset-submission</code></a> GitHub repo.
The submission scripts used for the kinase inhibitor and subsituted phenyl datasets are listed below.</p>
<ul>
<li><a href="https://github.com/openforcefield/qca-dataset-submission/tree/master/2019-11-27-kinase-inhibitor-optimization">Kinase inhibitor dataset</a></li>
<li><a href="https://github.com/openforcefield/qca-dataset-submission/tree/master/2019-07-25-phenyl-set">Subsituted phenyl dataset</a></li>
</ul>
<h4 id="details-on-qc-and-mm-torsion-scans.">5.1.2 Details on QC and MM torsion scans.</h4>
<p>All torsion scans were computed with the TorsionDrive <span class="citation" data-cites="PBjQ2b0k">[<a href="#ref-PBjQ2b0k" role="doc-biblioref">55</a>]</span> project, which makes choices of new constrained optimizations to evaluate.
The required constrained optimizations were then computed with the geomeTRIC <span class="citation" data-cites="zyjNwQ0g">[<a href="#ref-zyjNwQ0g" role="doc-biblioref">56</a>]</span> standalone geometry optimizer interfaced to the QCEngine <span class="citation" data-cites="1BYCAutAC">[<a href="#ref-1BYCAutAC" role="doc-biblioref">57</a>]</span> project.</p>
<p>To ensure a fair comparison between the QC and MM torsion scans, the only change in the torsion scan procedure was to switch out the program, which evaluated the gradient at each step in the geomeTRIC optimization.
For QC, gradients were computed at B3LYP-D3(BJ) / DZVP with the Psi4 <span class="citation" data-cites="YC6qkEYS">[<a href="#ref-YC6qkEYS" role="doc-biblioref">38</a>]</span> program.
Our choice of quantum chemical level of theory and basis set was based on benchmarks of quantum chemistry and density functional methods for the accuracy of conformational energies, such as <span class="citation" data-cites="18cMEuIQa">[<a href="#ref-18cMEuIQa" role="doc-biblioref">58</a>]</span>.
In these studies it was generally observed that B3LYP-D3(BJ) which includes an empirical dispersion correction <span class="citation" data-cites="1Jo5rmoB">[<a href="#ref-1Jo5rmoB" role="doc-biblioref">61</a>]</span> is roughly equivalent to MP2 and the wB97X-V functional with nonlocal dispersion <span class="citation" data-cites="bDaMuroD">[<a href="#ref-bDaMuroD" role="doc-biblioref">62</a>]</span> in terms of accuracy for conformational energies of small molecules (&lt;30 heavy atoms) in the complete basis set limit, i.e. 0.3-0.4 kcal/mol vs. CCSD(T)/CBS gold standard calculations.
Remarkably, when using the DZVP basis set <span class="citation" data-cites="17YEERBX3">[<a href="#ref-17YEERBX3" role="doc-biblioref">44</a>]</span> which is equivalent to 6-31G* in size and was optimized for DFT calculations, the accuracy of conformational energy calculations was unchanged compared to the much larger def2-TZVPD and def2-QZVP basis sets<span class="citation" data-cites="1CGEPtGug">[<a href="#ref-1CGEPtGug" role="doc-biblioref">63</a>]</span>.
After verification of the published results locally, we chose B3LYP-D3(BJ)/DZVP as the QM method of choice for torsion drives.
For molecular mechanics, gradients were run using OpenMM <span class="citation" data-cites="sJWPs71N">[<a href="#ref-sJWPs71N" role="doc-biblioref">64</a>]</span> with the OpenFF parsley Force Field (v1.0.0) <span class="citation" data-cites="j7nbtcyi">[<a href="#ref-j7nbtcyi" role="doc-biblioref">65</a>]</span>.</p>
<h3 id="calculating-bond-orders">5.2 Calculating Bond orders</h3>
<h4 id="am1-wbo-and-am1-elf10-wbo">5.2.1 AM1 WBO and AM1 ELF10 WBO</h4>
<p>To calculate AM1 ELF10 WBO, we used OpenEye’s QUACPAC toolkit <span class="citation" data-cites="OlXUfAPe">[<a href="#ref-OlXUfAPe" role="doc-biblioref">66</a>]</span> (OpenEye version 2019.Apr.2). The
ELF10 WBO comes along free after an AM1-BCC charge fitting procedure. For ELF10 WBOs generated in this paper, we used the <code>get_charges</code> function
in the <code>chemi.py</code> module in <code>fragmenter</code> versions v0.0.3 and v0.0.4. To calculate AM1 WBO for individual conformers, we used the <code>OEAssignPartialCharges</code>
with the <code>OECharges_AM1BCCSym</code> option from the QUACPAC toolkit for each conformer generated with Omega <span class="citation" data-cites="9rOEoE68">[<a href="#ref-9rOEoE68" role="doc-biblioref">67</a>]</span> (OpenEye version 2019.Apr.2)
which is called for the <code>get_charges</code> function.
For AM1 WBOs calculated to verify the results from the validation set, we generated conformers using the <code>generate_grid_conformer</code> function in the <code>chemi.py</code> module in <code>fragmenter</code> version v0.0.6.</p>
<p>We noticed that AM1 ELF10 WBOs can differ significantly across platforms (Linux vs Mac OS) for triple bonds. All results in this paper were calculated on a Linux machine. The
results might be different on a Mac OS.</p>
<h4 id="wiberg-bond-orders-in-psi4">5.2.2 Wiberg Bond Orders in Psi4</h4>
<p>Wiberg-Löwdin bond orders are calculated in Psi4 with the keyword <code>scf_properties: wiberg_lowdin_indices</code> using Psi4 version 1.3. All bond orders were computed during the torsion scan computations.</p>
<h3 id="datasets">5.3 Datasets</h3>
<h4 id="kinase-inhibitor-dataset">5.3.1 Kinase inhibitor dataset</h4>
<p>The kinase inhibitor dataset consists of 43 FDA approved kinase inhibitors (smi files is in the SI) with their Omega generated conformers (OpenEye veriso 2019.Apr.2, <code>generate_conformers</code>
function in the <code>chemi.py</code> module in <code>fragmenter</code> version v0.0.4). AM1 WBOs were calculated as described above, for all conformers of all 43 kinase inhibitors. B3LYP-D3(BJ) / DZVP Wiberg-Löwdin
bond orders were calculated for 9 kinase inhibitors and Omega generated conformers after a B3LYP-D3P(BJ) / DZVP geometry optimization. The DFT results are available on QCArchive
as an <code>OptimizationDataset</code> named <code>Kinase Inhibitors: WBO Distributions</code>.</p>
<p>The variance of the WBO distributions were calculated using the <code>numpy</code> <span class="citation" data-cites="OiM2HfsY">[<a href="#ref-OiM2HfsY" role="doc-biblioref">68</a>]</span> <code>var</code> function version 1.16.2 and their confidence intervals were calculated using arch <code>IIDBootsrap</code> function <span class="citation" data-cites="15oTe7N42">[<a href="#ref-15oTe7N42" role="doc-biblioref">69</a>]</span> version 4.8.1.
To calculate the correlation matrix, we calculated the Pearson correlation coefficient with the <code>numpy</code> <span class="citation" data-cites="OiM2HfsY">[<a href="#ref-OiM2HfsY" role="doc-biblioref">68</a>]</span> <code>corrcoef</code> function version 1.16.2.
Scripts and data used to generate and analyze this dataset are in
<a href="https://github.com/choderalab/fragmenter_data/tree/master/manuscript-figures/kinase_inhibitors_wbos">github.com/choderalab/fragmenter_data/manuscript-figures/kinase_inhibitors_wbos</a></p>
<h4 id="subsituted-phenyl-dataset">5.3.2 Subsituted phenyl dataset</h4>
<p>The substituted phenyl dataset consists of 3,458 substituted phenyl molecules where the substituents chosen to span a large range of electron donating and withdrawing groups.
We arrived at 3,200 molecules by attaching 26 different functional groups to 5 scaffolds (Figure ??, A) at the X<sub>1</sub> position, and then attach the 26 functional group (and H) at the X<sub>2</sub> position for a total of 133 molecules per functional group (26 * 5 + 3 (for molecules with H at the X<sub>2</sub> position)).
The AM1 ELF10 WBOs were calculated as described above.</p>
<p>We selected molecules for QC torsion scans as follows:
1. From the 26 functional groups, we only selected molecules from 18 functional groups, skipping X<sub>1</sub>s that either did not have a rotatable bond (fluorine, chlorine, iodine, bromine, nitrile, oxygen), were too congested (triflouromethyl, trimethylamonium) or where the WBOs on the bonds attaching X<sub>1</sub> to the phenyl ring did not change much
with different chemical group at the X<sub>2</sub> position (methyl).
2. For the 18 functional groups, we chose molecules that were evenly spaced along the WBO range of that functional group, up to 15 molecules.
While all the skipped functional groups for X<sub>1</sub> were allowed to be at X<sub>2</sub>, we did not include the negative oxygen at X<sub>2</sub> because OpenFF have not yet benchmarked the level of theory to use for anions.
3. After selection, we had 140 molecules that we submitted to QCArchive for both QC and MM torsion scan.
The dataset is available on QCArchive as a <code>TorsionDriveDataset</code> named <code>OpenFF Subsituted Phenyl Set 1</code>.
This dataset also includes the biphenyl torsion scans shown in Figure <a href="#fig:biphenyls">1</a>.</p>
<p>Scripts used to generate and analyze this dataset can be found in <a href="https://github.com/choderalab/fragmenter_data/tree/master/phenyl_benchmark">github.com/choderalab/fragmenter_data/phenyl_benchmark</a></p>
<p>There is another subsituted phenyl set on QCArchive whose results are not shown in this paper because it was run with a different level of theory as the default OpenFF level of theory, included anions which we did not yet decide how to handle and did not have good coverage of WBO ranges.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-aLuXbQN8">
<p>1. <strong>Molecular Dynamics Simulation for All</strong><br />
Scott A. Hollingsworth, Ron O. Dror<br />
<em>Neuron</em> (2018-09) <a href="https://doi.org/gd8gf9">https://doi.org/gd8gf9</a><br />
DOI: <a href="https://doi.org/10.1016/j.neuron.2018.08.011">10.1016/j.neuron.2018.08.011</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30236283">30236283</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6209097">PMC6209097</a></p>
</div>
<div id="ref-qhEZ9qxK">
<p>2. <strong>The dynamic conformational landscape of the protein methyltransferase SETD8</strong><br />
Shi Chen, Rafal P Wiewiora, Fanwang Meng, Nicolas Babault, Anqi Ma, Wenyu Yu, Kun Qian, Hao Hu, Hua Zou, Junyi Wang, … Minkui Luo<br />
<em>eLife</em> (2019-05-13) <a href="https://doi.org/ggfkmq">https://doi.org/ggfkmq</a><br />
DOI: <a href="https://doi.org/10.7554/elife.45403">10.7554/elife.45403</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31081496">31081496</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6579520">PMC6579520</a></p>
</div>
<div id="ref-9EY3maYh">
<p>3. <strong>A Functional Selectivity Mechanism at the Serotonin-2A GPCR Involves Ligand-Dependent Conformations of Intracellular Loop 2</strong><br />
Jose Manuel Perez-Aguilar, Jufang Shan, Michael V. LeVine, George Khelashvili, Harel Weinstein<br />
<em>Journal of the American Chemical Society</em> (2014-10-31) <a href="https://doi.org/f6qknr">https://doi.org/f6qknr</a><br />
DOI: <a href="https://doi.org/10.1021/ja508394x">10.1021/ja508394x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25314362">25314362</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4235374">PMC4235374</a></p>
</div>
<div id="ref-mbmV3Fg8">
<p>4. <strong>Toward Learned Chemical Perception of Force Field Typing Rules</strong><br />
Camila Zanette, Caitlin C. Bannan, Christopher I. Bayly, Josh Fass, Michael K. Gilson, Michael R. Shirts, John D. Chodera, David L. Mobley<br />
<em>Journal of Chemical Theory and Computation</em> (2018-12-04) <a href="https://doi.org/gft4hf">https://doi.org/gft4hf</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.8b00821">10.1021/acs.jctc.8b00821</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30512951">30512951</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6467725">PMC6467725</a></p>
</div>
<div id="ref-18mvPmwwB">
<p>5. <strong>OPLS3: A Force Field Providing Broad Coverage of Drug-like Small Molecules and Proteins</strong><br />
Edward Harder, Wolfgang Damm, Jon Maple, Chuanjie Wu, Mark Reboul, Jin Yu Xiang, Lingle Wang, Dmitry Lupyan, Markus K. Dahlgren, Jennifer L. Knight, … Richard A. Friesner<br />
<em>Journal of Chemical Theory and Computation</em> (2015-12) <a href="https://doi.org/f76wpm">https://doi.org/f76wpm</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.5b00864">10.1021/acs.jctc.5b00864</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26584231">26584231</a></p>
</div>
<div id="ref-11Z8pXbEW">
<p>6. <strong>CHARMM: A program for macromolecular energy, minimization, and dynamics calculations</strong><br />
Bernard R. Brooks, Robert E. Bruccoleri, Barry D. Olafson, David J. States, S. Swaminathan, Martin Karplus<br />
<em>Journal of Computational Chemistry</em> (1983-Summer) <a href="https://doi.org/bqh7f2">https://doi.org/bqh7f2</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.540040211">10.1002/jcc.540040211</a></p>
</div>
<div id="ref-1GDaakPWY">
<p>7. <strong>A Second Generation Force Field for the Simulation of Proteins, Nucleic Acids, and Organic Molecules</strong><br />
Wendy D. Cornell, Piotr Cieplak, Christopher I. Bayly, Ian R. Gould, Kenneth M. Merz, David M. Ferguson, David C. Spellmeyer, Thomas Fox, James W. Caldwell, Peter A. Kollman<br />
<em>Journal of the American Chemical Society</em> (1995-05) <a href="https://doi.org/dbzh27">https://doi.org/dbzh27</a><br />
DOI: <a href="https://doi.org/10.1021/ja00124a002">10.1021/ja00124a002</a></p>
</div>
<div id="ref-Mi3Ujd07">
<p>8. <strong>Development and Testing of the OPLS All-Atom Force Field on Conformational Energetics and Properties of Organic Liquids</strong><br />
William L. Jorgensen, David S. Maxwell, Julian Tirado-Rives<br />
<em>Journal of the American Chemical Society</em> (1996-01) <a href="https://doi.org/fvftxj">https://doi.org/fvftxj</a><br />
DOI: <a href="https://doi.org/10.1021/ja9621760">10.1021/ja9621760</a></p>
</div>
<div id="ref-i8ZZOoVS">
<p>9. <strong>Automated conformational energy fitting for force-field development</strong><br />
Olgun Guvench, Alexander D. MacKerell<br />
<em>Journal of Molecular Modeling</em> (2008-05-06) <a href="https://doi.org/bzphqw">https://doi.org/bzphqw</a><br />
DOI: <a href="https://doi.org/10.1007/s00894-008-0305-0">10.1007/s00894-008-0305-0</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18458967">18458967</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2864003">PMC2864003</a></p>
</div>
<div id="ref-fpE0Y69s">
<p>10. <strong>Torsional Energy Barriers of Biaryls Could Be Predicted by Electron Richness/Deficiency of Aromatic Rings; Advancement of Molecular Mechanics toward Atom-Type Independence</strong><br />
Wanlei Wei, Candide Champion, Zhaomin Liu, Stephen J. Barigye, Paul Labute, Nicolas Moitessier<br />
<em>Journal of Chemical Information and Modeling</em> (2019-08-20) <a href="https://doi.org/ggh677">https://doi.org/ggh677</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.9b00585">10.1021/acs.jcim.9b00585</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31430147">31430147</a></p>
</div>
<div id="ref-AWj8hnbG">
<p>11. <strong>Atom Types Independent Molecular Mechanics Method for Predicting the Conformational Energy of Small Molecules</strong><br />
Zhaomin Liu, Stephen J. Barigye, Moeed Shahamat, Paul Labute, Nicolas Moitessier<br />
<em>Journal of Chemical Information and Modeling</em> (2018-01-05) <a href="https://doi.org/gcxm7s">https://doi.org/gcxm7s</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.7b00645">10.1021/acs.jcim.7b00645</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29253333">29253333</a></p>
</div>
<div id="ref-Vx7ALVw2">
<p>12. <strong>Hyperconjugation not steric repulsion leads to the staggered structure of ethane</strong><br />
Vojislava Pophristic, Lionel Goodman<br />
<em>Nature</em> (2001-05) <a href="https://doi.org/dff6j7">https://doi.org/dff6j7</a><br />
DOI: <a href="https://doi.org/10.1038/35079036">10.1038/35079036</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/11385566">11385566</a></p>
</div>
<div id="ref-1DVkKVgPL">
<p>13. <strong>Gauche Effect in 1,2-Difluoroethane. Hyperconjugation, Bent Bonds, Steric Repulsion</strong><br />
Lionel Goodman, Hongbing Gu, Vojislava Pophristic<br />
<em>The Journal of Physical Chemistry A</em> (2005-02) <a href="https://doi.org/dtg782">https://doi.org/dtg782</a><br />
DOI: <a href="https://doi.org/10.1021/jp046290d">10.1021/jp046290d</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16833433">16833433</a></p>
</div>
<div id="ref-13fQUQyf9">
<p>14. <strong>Accuracy evaluation and addition of improved dihedral parameters for the MMFF94s</strong><br />
Joel Wahl, Joel Freyss, Modest von Korff, Thomas Sander<br />
<em>Journal of Cheminformatics</em> (2019-08-07) <a href="https://doi.org/gf6rz2">https://doi.org/gf6rz2</a><br />
DOI: <a href="https://doi.org/10.1186/s13321-019-0371-6">10.1186/s13321-019-0371-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31392432">31392432</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6686419">PMC6686419</a></p>
</div>
<div id="ref-thaStBUv">
<p>15. <strong>Automated Force Field Parameterization for Nonpolarizable and Polarizable Atomic Models Based on Ab Initio Target Data</strong><br />
Lei Huang, Benoît Roux<br />
<em>Journal of Chemical Theory and Computation</em> (2013-07-25) <a href="https://doi.org/f47z2j">https://doi.org/f47z2j</a><br />
DOI: <a href="https://doi.org/10.1021/ct4003477">10.1021/ct4003477</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24223528">24223528</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3819940">PMC3819940</a></p>
</div>
<div id="ref-jcQ6P6uD">
<p>16. <strong>Hierarchical atom type definitions and extensible all-atom force fields</strong><br />
Zhao Jin, Chunwei Yang, Fenglei Cao, Feng Li, Zhifeng Jing, Long Chen, Zhe Shen, Liang Xin, Sijia Tong, Huai Sun<br />
<em>Journal of Computational Chemistry</em> (2016-03-15) <a href="https://doi.org/f8bmzq">https://doi.org/f8bmzq</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.24244">10.1002/jcc.24244</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26537332">26537332</a></p>
</div>
<div id="ref-VYvjE9wM">
<p>17. <strong>Automation of the CHARMM General Force Field (CGenFF) I: Bond Perception and Atom Typing</strong><br />
K. Vanommeslaeghe, A. D. MacKerell<br />
<em>Journal of Chemical Information and Modeling</em> (2012-11-28) <a href="https://doi.org/f4gn45">https://doi.org/f4gn45</a><br />
DOI: <a href="https://doi.org/10.1021/ci300363c">10.1021/ci300363c</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23146088">23146088</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3528824">PMC3528824</a></p>
</div>
<div id="ref-Z8BM4e7W">
<p>18. <strong>CHARMM general force field: A force field for drug-like molecules compatible with the CHARMM all-atom additive biological force fields</strong><br />
K. Vanommeslaeghe, E. Hatcher, C. Acharya, S. Kundu, S. Zhong, J. Shim, E. Darian, O. Guvench, P. Lopes, I. Vorobyov, A. D. Mackerell<br />
<em>Journal of Computational Chemistry</em> (2009) <a href="https://doi.org/fgkjs4">https://doi.org/fgkjs4</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.21367">10.1002/jcc.21367</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19575467">19575467</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2888302">PMC2888302</a></p>
</div>
<div id="ref-YmRgHfeU">
<p>19. <strong>Development and testing of a general amber force field</strong><br />
Junmei Wang, Romain M. Wolf, James W. Caldwell, Peter A. Kollman, David A. Case<br />
<em>Journal of Computational Chemistry</em> (2004-07-15) <a href="https://doi.org/cdmcnb">https://doi.org/cdmcnb</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.20035">10.1002/jcc.20035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15116359">15116359</a></p>
</div>
<div id="ref-1F62Qo2RU">
<p>20. <strong>QUBEKit: Automating the Derivation of Force Field Parameters from Quantum Mechanics</strong><br />
Joshua T. Horton, Alice E. A. Allen, Leela S. Dodda, Daniel J. Cole<br />
<em>Journal of Chemical Information and Modeling</em> (2019-02-11) <a href="https://doi.org/gfvvf9">https://doi.org/gfvvf9</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.8b00767">10.1021/acs.jcim.8b00767</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30742438">30742438</a></p>
</div>
<div id="ref-JJX4iDz2">
<p>21. <strong>Characterization of Biaryl Torsional Energetics and its Treatment in OPLS All-Atom Force Fields</strong><br />
Markus K. Dahlgren, Patric Schyman, Julian Tirado-Rives, William L. Jorgensen<br />
<em>Journal of Chemical Information and Modeling</em> (2013-05-13) <a href="https://doi.org/f4xn5p">https://doi.org/f4xn5p</a><br />
DOI: <a href="https://doi.org/10.1021/ci4001597">10.1021/ci4001597</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23621692">23621692</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3707137">PMC3707137</a></p>
</div>
<div id="ref-1HYTTY1PU">
<p>22. <strong>Escaping Atom Types in Force Fields Using Direct Chemical Perception</strong><br />
David L. Mobley, Caitlin C. Bannan, Andrea Rizzi, Christopher I. Bayly, John D. Chodera, Victoria T. Lim, Nathan M. Lim, Kyle A. Beauchamp, David R. Slochower, Michael R. Shirts, … Peter K. Eastman<br />
<em>Journal of Chemical Theory and Computation</em> (2018-10-11) <a href="https://doi.org/gffnf3">https://doi.org/gffnf3</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.8b00640">10.1021/acs.jctc.8b00640</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30351006">30351006</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6245550">PMC6245550</a></p>
</div>
<div id="ref-15wrEpSer">
<p>23. <strong>Elucidating Hyperconjugation from Electronegativity to Predict Drug Conformational Energy in a High Throughput Manner</strong><br />
Zhaomin Liu, Joshua Pottel, Moeed Shahamat, Anna Tomberg, Paul Labute, Nicolas Moitessier<br />
<em>Journal of Chemical Information and Modeling</em> (2016-04-12) <a href="https://doi.org/f8kztt">https://doi.org/f8kztt</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.6b00012">10.1021/acs.jcim.6b00012</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27028941">27028941</a></p>
</div>
<div id="ref-16GNoHAFY">
<p>24. <strong>Atom Type Independent Modeling of the Conformational Energy of Benzylic, Allylic, and Other Bonds Adjacent to Conjugated Systems</strong><br />
Candide Champion, Stephen J. Barigye, Wanlei Wei, Zhaomin Liu, Paul Labute, Nicolas Moitessier<br />
<em>Journal of Chemical Information and Modeling</em> (2019-10-07) <a href="https://doi.org/ggh676">https://doi.org/ggh676</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.9b00581">10.1021/acs.jcim.9b00581</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31589815">31589815</a></p>
</div>
<div id="ref-ZAAgsuwk">
<p>25. <strong>The Dependence of Interatomic Distance on Single Bond-Double Bond Resonance <sup>1</sup></strong><br />
Linus Pauling, L. O. Brockway, J. Y. Beach<br />
<em>Journal of the American Chemical Society</em> (1935-12) <a href="https://doi.org/fswdwx">https://doi.org/fswdwx</a><br />
DOI: <a href="https://doi.org/10.1021/ja01315a105">10.1021/ja01315a105</a></p>
</div>
<div id="ref-QY82Gb6W">
<p>26. <strong>The electronic structure of some polyenes and aromatic molecules. VII. Bonds of fractional order by the molecular orbital method</strong><br />
Proceedings of the Royal Society of London. Series A. Mathematical and Physical Sciences<br />
(1997-01) <a href="https://doi.org/bnqx76">https://doi.org/bnqx76</a><br />
DOI: <a href="https://doi.org/10.1098/rspa.1939.0006">10.1098/rspa.1939.0006</a></p>
</div>
<div id="ref-9WKaagH4">
<p>27. <strong>Electronic Population Analysis on LCAO–MO Molecular Wave Functions. I</strong><br />
R. S. Mulliken<br />
<em>The Journal of Chemical Physics</em> (1955-10) <a href="https://doi.org/d24mtt">https://doi.org/d24mtt</a><br />
DOI: <a href="https://doi.org/10.1063/1.1740588">10.1063/1.1740588</a></p>
</div>
<div id="ref-BES2Ksiq">
<p>28. <strong>Application of the pople-santry-segal CNDO method to the cyclopropylcarbinyl and cyclobutyl cation and to bicyclobutane</strong><br />
K. B. Wiberg<br />
<em>Tetrahedron</em> (1968-01) <a href="https://doi.org/fvwkkh">https://doi.org/fvwkkh</a><br />
DOI: <a href="https://doi.org/10.1016/0040-4020(68)88057-3">10.1016/0040-4020(68)88057-3</a></p>
</div>
<div id="ref-8uYd8cxr">
<p>29. <strong>Charge, bond order and valence in the AB initio SCF theory</strong><br />
I. Mayer<br />
<em>Chemical Physics Letters</em> (1983-05) <a href="https://doi.org/b5m8r5">https://doi.org/b5m8r5</a><br />
DOI: <a href="https://doi.org/10.1016/0009-2614(83)80005-0">10.1016/0009-2614(83)80005-0</a></p>
</div>
<div id="ref-vfRrdWeG">
<p>30. <strong>A maximum bond order principle</strong><br />
Karl Jug<br />
<em>Journal of the American Chemical Society</em> (1977-11) <a href="https://doi.org/dvfv63">https://doi.org/dvfv63</a><br />
DOI: <a href="https://doi.org/10.1021/ja00466a009">10.1021/ja00466a009</a></p>
</div>
<div id="ref-11ndqhfsy">
<p>31. <strong>Bond Orders of Homonuclear Diatomic Molecules</strong><br />
Peter Politzer<br />
<em>The Journal of Chemical Physics</em> (1969-03-15) <a href="https://doi.org/d28qgt">https://doi.org/d28qgt</a><br />
DOI: <a href="https://doi.org/10.1063/1.1671457">10.1063/1.1671457</a></p>
</div>
<div id="ref-mgZozbxT">
<p>32. <strong>The MMP2 calculational method</strong><br />
Joseph T. Sprague, Julia C. Tai, Young Yuh, Norman L. Allinger<br />
<em>Journal of Computational Chemistry</em> (1987-07) <a href="https://doi.org/dpjxnn">https://doi.org/dpjxnn</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.540080504">10.1002/jcc.540080504</a></p>
</div>
<div id="ref-F7LEPYAx">
<p>33. <strong>Molecular mechanics (MM3) calculations on conjugated hydrocarbons</strong><br />
Norman L. Allinger, Fanbing Li, Liqun Yan, Julia C. Tai<br />
<em>Journal of Computational Chemistry</em> (1990-08) <a href="https://doi.org/fs344c">https://doi.org/fs344c</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.540110709">10.1002/jcc.540110709</a></p>
</div>
<div id="ref-waPeRcgV">
<p>34. <strong>A new force field for molecular mechanical simulation of nucleic acids and proteins</strong><br />
Scott J. Weiner, Peter A. Kollman, David A. Case, U. Chandra Singh, Caterina Ghio, Guliano Alagona, Salvatore Profeta, Paul Weiner<br />
<em>Journal of the American Chemical Society</em> (1984-02) <a href="https://doi.org/dcvwwx">https://doi.org/dcvwwx</a><br />
DOI: <a href="https://doi.org/10.1021/ja00315a051">10.1021/ja00315a051</a></p>
</div>
<div id="ref-QlopodHU">
<p>35. <strong>Bond order and valence indices: A personal account</strong><br />
I. Mayer<br />
<em>Journal of Computational Chemistry</em> (2007-01-15) <a href="https://doi.org/ddssrj">https://doi.org/ddssrj</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.20494">10.1002/jcc.20494</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17066501">17066501</a></p>
</div>
<div id="ref-tnSqySMX">
<p>36. <strong>Development and use of quantum mechanical molecular models. 76. AM1: a new general purpose quantum mechanical molecular model</strong><br />
Michael J. S. Dewar, Eve G. Zoebisch, Eamonn F. Healy, James J. P. Stewart<br />
<em>Journal of the American Chemical Society</em> (1985-06) <a href="https://doi.org/fd8bwp">https://doi.org/fd8bwp</a><br />
DOI: <a href="https://doi.org/10.1021/ja00299a024">10.1021/ja00299a024</a></p>
</div>
<div id="ref-dgUvu4tX">
<p>37. <strong>Optimization of parameters for semiempirical methods I. Method</strong><br />
James J. P. Stewart<br />
<em>Journal of Computational Chemistry</em> (1989-03) <a href="https://doi.org/cg5pck">https://doi.org/cg5pck</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.540100208">10.1002/jcc.540100208</a></p>
</div>
<div id="ref-YC6qkEYS">
<p>38. <strong>1.1: An Open-Source Electronic Structure Program Emphasizing Automation, Advanced Libraries, and Interoperability</strong><br />
Robert M. Parrish, Lori A. Burns, Daniel G. A. Smith, Andrew C. Simmonett, A. Eugene DePrince, Edward G. Hohenstein, Uğur Bozkaya, Alexander Yu. Sokolov, Roberto Di Remigio, Ryan M. Richard, … C. David Sherrill<br />
<em>Journal of Chemical Theory and Computation</em> (2017-06-06) <a href="https://doi.org/gcz64j">https://doi.org/gcz64j</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.7b00174">10.1021/acs.jctc.7b00174</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28489372">28489372</a></p>
</div>
<div id="ref-K8bdDsCu">
<p>39. <strong>On the Non‐Orthogonality Problem Connected with the Use of Atomic Wave Functions in the Theory of Molecules and Crystals</strong><br />
Per‐Olov Löwdin<br />
<em>The Journal of Chemical Physics</em> (1950-03) <a href="https://doi.org/dj2c35">https://doi.org/dj2c35</a><br />
DOI: <a href="https://doi.org/10.1063/1.1747632">10.1063/1.1747632</a></p>
</div>
<div id="ref-1BEyXJiMo">
<p>40. <strong>On the quantum theory of valence and bonding from the ab intio standpoint</strong><br />
Mario A. Natiello, Jorge A. Medrano<br />
<em>Chemical Physics Letters</em> (1984-03) <a href="https://doi.org/bdfk5f">https://doi.org/bdfk5f</a><br />
DOI: <a href="https://doi.org/10.1016/0009-2614(84)85645-6">10.1016/0009-2614(84)85645-6</a></p>
</div>
<div id="ref-r3xLNmRT">
<p>41. <strong>Can we treat ab initio atomic charges and bond orders as conformation-independent electronic structure descriptors?</strong><br />
T. Yu. Nikolaienko, L. A. Bulavin, D. M. Hovorun<br />
<em>RSC Advances</em> (2016) <a href="https://doi.org/gf66tp">https://doi.org/gf66tp</a><br />
DOI: <a href="https://doi.org/10.1039/c6ra17055b">10.1039/c6ra17055b</a></p>
</div>
<div id="ref-hIon1SaZ">
<p>42. <strong>Resonance interactions in acyclic systems. 3. Formamide internal rotation revisited. Charge and energy redistribution along the C-N bond rotational pathway</strong><br />
Kenneth B. Wiberg, Curt M. Breneman<br />
<em>Journal of the American Chemical Society</em> (1992-01) <a href="https://doi.org/c85zcc">https://doi.org/c85zcc</a><br />
DOI: <a href="https://doi.org/10.1021/ja00029a005">10.1021/ja00029a005</a></p>
</div>
<div id="ref-1G1ccz54w">
<p>43. <strong>Conformer Generation with OMEGA: Algorithm and Validation Using High Quality Structures from the Protein Databank and Cambridge Structural Database</strong><br />
Paul C. D. Hawkins, A. Geoffrey Skillman, Gregory L. Warren, Benjamin A. Ellingson, Matthew T. Stahl<br />
<em>Journal of Chemical Information and Modeling</em> (2010-03-17) <a href="https://doi.org/d4rb6g">https://doi.org/d4rb6g</a><br />
DOI: <a href="https://doi.org/10.1021/ci100031x">10.1021/ci100031x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20235588">20235588</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2859685">PMC2859685</a></p>
</div>
<div id="ref-17YEERBX3">
<p>44. <strong>Optimization of Gaussian-type basis sets for local spin density functional calculations. Part I. Boron through neon, optimization technique and validation</strong><br />
Nathalie Godbout, Dennis R. Salahub, Jan Andzelm, Erich Wimmer<br />
<em>Canadian Journal of Chemistry</em> (1992-02-01) <a href="https://doi.org/c78qjn">https://doi.org/c78qjn</a><br />
DOI: <a href="https://doi.org/10.1139/v92-079">10.1139/v92-079</a></p>
</div>
<div id="ref-flKRoCNc">
<p>45. <strong>Density‐functional thermochemistry. III. The role of exact exchange</strong><br />
Axel D. Becke<br />
<em>The Journal of Chemical Physics</em> (1993-04) <a href="https://doi.org/d2fq9j">https://doi.org/d2fq9j</a><br />
DOI: <a href="https://doi.org/10.1063/1.464913">10.1063/1.464913</a></p>
</div>
<div id="ref-YPDz9hB3">
<p>46. <strong>A consistent and accurate <em>ab initio</em> parametrization of density functional dispersion correction (DFT-D) for the 94 elements H-Pu</strong><br />
Stefan Grimme, Jens Antony, Stephan Ehrlich, Helge Krieg<br />
<em>The Journal of Chemical Physics</em> (2010-04-21) <a href="https://doi.org/bnt82x">https://doi.org/bnt82x</a><br />
DOI: <a href="https://doi.org/10.1063/1.3382344">10.1063/1.3382344</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20423165">20423165</a></p>
</div>
<div id="ref-CqesmIsy">
<p>47. <strong>Effect of the damping function in dispersion corrected density functional theory</strong><br />
Stefan Grimme, Stephan Ehrlich, Lars Goerigk<br />
<em>Journal of Computational Chemistry</em> (2011-05) <a href="https://doi.org/b3ztst">https://doi.org/b3ztst</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.21759">10.1002/jcc.21759</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21370243">21370243</a></p>
</div>
<div id="ref-2hGWnSJL">
<p>48. <strong>Benchmarking Commercial Conformer Ensemble Generators</strong><br />
Nils-Ole Friedrich, Christina de Bruyn Kops, Florian Flachsenberg, Kai Sommer, Matthias Rarey, Johannes Kirchmair<br />
<em>Journal of Chemical Information and Modeling</em> (2017-10-18) <a href="https://doi.org/gb4v2v">https://doi.org/gb4v2v</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jcim.7b00505">10.1021/acs.jcim.7b00505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28967749">28967749</a></p>
</div>
<div id="ref-14qKIxxbD">
<p>49. <strong>Resonance interactions in acyclic systems</strong><br />
K. B. Wiberg, C. M. Breneman, K. E. Laidig, R. E. Rosenberg<br />
<em>Pure and Applied Chemistry</em> (1989-01-01) <a href="https://doi.org/c6rd43">https://doi.org/c6rd43</a><br />
DOI: <a href="https://doi.org/10.1351/pac198961040635">10.1351/pac198961040635</a></p>
</div>
<div id="ref-16hZpgNca">
<p>50. <strong>QUACPAC 2.0.2.2 — Applications, v2019.Nov.2</strong><a href="https://docs.eyesopen.com/applications/quacpac/index.html">https://docs.eyesopen.com/applications/quacpac/index.html</a></p>
</div>
<div id="ref-LxrgIkt0">
<p>51. <strong>Fast, efficient generation of high-quality atomic charges. AM1-BCC model: II. Parameterization and validation</strong><br />
Araz Jakalian, David B. Jack, Christopher I. Bayly<br />
<em>Journal of Computational Chemistry</em> (2002-10-18) <a href="https://doi.org/cktk6g">https://doi.org/cktk6g</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.10128">10.1002/jcc.10128</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12395429">12395429</a></p>
</div>
<div id="ref-3z27USyZ">
<p>52. <strong>The Effect of Structure upon the Reactions of Organic Compounds. Benzene Derivatives</strong><br />
Louis P. Hammett<br />
<em>Journal of the American Chemical Society</em> (1937-01) <a href="https://doi.org/dz8d4r">https://doi.org/dz8d4r</a><br />
DOI: <a href="https://doi.org/10.1021/ja01280a022">10.1021/ja01280a022</a></p>
</div>
<div id="ref-1EJC88xIC">
<p>53. <strong>A survey of Hammett substituent constants and resonance and field parameters</strong><br />
Corwin. Hansch, A. Leo, R. W. Taft<br />
<em>Chemical Reviews</em> (1991-03) <a href="https://doi.org/dwwqnk">https://doi.org/dwwqnk</a><br />
DOI: <a href="https://doi.org/10.1021/cr00002a004">10.1021/cr00002a004</a></p>
</div>
<div id="ref-QS8UBx4">
<p>54. <strong>Perspective: Computational chemistry software and its advancement as illustrated through three grand challenge cases for molecular science</strong><br />
Anna Krylov, Theresa L. Windus, Taylor Barnes, Eliseo Marin-Rimoldi, Jessica A. Nash, Benjamin Pritchard, Daniel G. A. Smith, Doaa Altarawy, Paul Saxe, Cecilia Clementi, … Teresa Head-Gordon<br />
<em>The Journal of Chemical Physics</em> (2018-11-14) <a href="https://doi.org/ggjxmj">https://doi.org/ggjxmj</a><br />
DOI: <a href="https://doi.org/10.1063/1.5052551">10.1063/1.5052551</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30441927">30441927</a></p>
</div>
<div id="ref-PBjQ2b0k">
<p>55. <strong>lpwgroup/torsiondrive</strong><br />
lpwgroup<br />
(2020-02-24) <a href="https://github.com/lpwgroup/torsiondrive">https://github.com/lpwgroup/torsiondrive</a></p>
</div>
<div id="ref-zyjNwQ0g">
<p>56. <strong>leeping/geomeTRIC</strong><br />
Lee-Ping<br />
(2020-04-19) <a href="https://github.com/leeping/geomeTRIC">https://github.com/leeping/geomeTRIC</a></p>
</div>
<div id="ref-1BYCAutAC">
<p>57. <strong>MolSSI/QCEngine</strong><br />
The Molecular Science Software Institute<br />
(2020-03-04) <a href="https://github.com/MolSSI/QCEngine">https://github.com/MolSSI/QCEngine</a></p>
</div>
<div id="ref-18cMEuIQa">
<p>58. <strong>Benchmark <em>ab Initio</em> Conformational Energies for the Proteinogenic Amino Acids through Explicitly Correlated Methods. Assessment of Density Functional Methods</strong><br />
Manoj K. Kesharwani, Amir Karton, Jan M. L. Martin<br />
<em>Journal of Chemical Theory and Computation</em> (2015-12-22) <a href="https://doi.org/f764qf">https://doi.org/f764qf</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.5b01066">10.1021/acs.jctc.5b01066</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26653705">26653705</a></p>
</div>
<div id="ref-5r70efN6">
<p>59. <strong>Toward Accurate Conformational Energies of Smaller Peptides and Medium-Sized Macrocycles: MPCONF196 Benchmark Energy Data Set</strong><br />
Jan Řezáč, Daniel Bím, Ondrej Gutten, Lubomír Rulíšek<br />
<em>Journal of Chemical Theory and Computation</em> (2018-02-20) <a href="https://doi.org/gc7x27">https://doi.org/gc7x27</a><br />
DOI: <a href="https://doi.org/10.1021/acs.jctc.7b01074">10.1021/acs.jctc.7b01074</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29461829">29461829</a></p>
</div>
<div id="ref-W5W98nc6">
<p>60. <strong>A look at the density functional theory zoo with the advanced GMTKN55 database for general main group thermochemistry, kinetics and noncovalent interactions</strong><br />
Lars Goerigk, Andreas Hansen, Christoph Bauer, Stephan Ehrlich, Asim Najibi, Stefan Grimme<br />
<em>Physical Chemistry Chemical Physics</em> (2017) <a href="https://doi.org/ggk3cz">https://doi.org/ggk3cz</a><br />
DOI: <a href="https://doi.org/10.1039/c7cp04913g">10.1039/c7cp04913g</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29110012">29110012</a></p>
</div>
<div id="ref-1Jo5rmoB">
<p>61. <strong>Exchange-hole dipole moment and the dispersion interaction revisited</strong><br />
Axel D. Becke, Erin R. Johnson<br />
<em>The Journal of Chemical Physics</em> (2007-10-21) <a href="https://doi.org/fqf57s">https://doi.org/fqf57s</a><br />
DOI: <a href="https://doi.org/10.1063/1.2795701">10.1063/1.2795701</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17949133">17949133</a></p>
</div>
<div id="ref-bDaMuroD">
<p>62. <strong>ωB97X-V: A 10-parameter, range-separated hybrid, generalized gradient approximation density functional with nonlocal correlation, designed by a survival-of-the-fittest strategy</strong><br />
Narbe Mardirossian, Martin Head-Gordon<br />
<em>Physical Chemistry Chemical Physics</em> (2014) <a href="https://doi.org/gf9n3d">https://doi.org/gf9n3d</a><br />
DOI: <a href="https://doi.org/10.1039/c3cp54374a">10.1039/c3cp54374a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24430168">24430168</a></p>
</div>
<div id="ref-1CGEPtGug">
<p>63. <strong>Balanced basis sets of split valence, triple zeta valence and quadruple zeta valence quality for H to Rn: Design and assessment of accuracy</strong><br />
Florian Weigend, Reinhart Ahlrichs<br />
<em>Physical Chemistry Chemical Physics</em> (2005) <a href="https://doi.org/b56gv3">https://doi.org/b56gv3</a><br />
DOI: <a href="https://doi.org/10.1039/b508541a">10.1039/b508541a</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16240044">16240044</a></p>
</div>
<div id="ref-sJWPs71N">
<p>64. <strong>OpenMM 7: Rapid development of high performance algorithms for molecular dynamics</strong><br />
Peter Eastman, Jason Swails, John D. Chodera, Robert T. McGibbon, Yutong Zhao, Kyle A. Beauchamp, Lee-Ping Wang, Andrew C. Simmonett, Matthew P. Harrigan, Chaya D. Stern, … Vijay S. Pande<br />
<em>PLOS Computational Biology</em> (2017-07-26) <a href="https://doi.org/gbppkv">https://doi.org/gbppkv</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1005659">10.1371/journal.pcbi.1005659</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28746339">28746339</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5549999">PMC5549999</a></p>
</div>
<div id="ref-j7nbtcyi">
<p>65. <strong>openforcefield/openforcefields: Version 1.0.0 “Parsley”</strong><br />
Jeff Wagner<br />
<em>Zenodo</em> (2019-10-12) <a href="https://doi.org/ggjgjk">https://doi.org/ggjgjk</a><br />
DOI: <a href="https://doi.org/10.5281/zenodo.3483227">10.5281/zenodo.3483227</a></p>
</div>
<div id="ref-OlXUfAPe">
<p>66. <strong>QUACPAC | Tautomer / Protomer Enumeration &amp; Charge Assignment | Database Preparation</strong><br />
OpenEye Scientific Software<br />
<a href="https://www.eyesopen.com/quacpac">https://www.eyesopen.com/quacpac</a></p>
</div>
<div id="ref-9rOEoE68">
<p>67. <strong>OMEGA | Multi-Conformer Structure Databases | Database Preparation | OpenEye Scientific</strong><br />
OpenEye Scientific Software<br />
<a href="https://www.eyesopen.com/omega">https://www.eyesopen.com/omega</a></p>
</div>
<div id="ref-OiM2HfsY">
<p>68. <strong>The NumPy Array: A Structure for Efficient Numerical Computation</strong><br />
Stéfan van der Walt, S Chris Colbert, Gaël Varoquaux<br />
<em>Computing in Science &amp; Engineering</em> (2011-03) <a href="https://doi.org/d8k4p9">https://doi.org/d8k4p9</a><br />
DOI: <a href="https://doi.org/10.1109/mcse.2011.37">10.1109/mcse.2011.37</a></p>
</div>
<div id="ref-15oTe7N42">
<p>69. <strong>bashtage/arch: Release 4.8.1</strong><br />
Kevin Sheppard, Stanislav Khrapov, Gábor Lipták, Rob Capellini, Esvhd, Hugle, JPN, Xavier RENE-CORAIL, Michael E. Rose, Jbrockmendel<br />
<em>Zenodo</em> (2019-03-28) <a href="https://doi.org/ggjfwh">https://doi.org/ggjfwh</a><br />
DOI: <a href="https://doi.org/10.5281/zenodo.2613877">10.5281/zenodo.2613877</a></p>
</div>
<div id="ref-19YdfyNRY">
<p>70. <strong>Approximate Self‐Consistent Molecular Orbital Theory. I. Invariant Procedures</strong><br />
J. A. Pople, D. P. Santry, G. A. Segal<br />
<em>The Journal of Chemical Physics</em> (1965-11-15) <a href="https://doi.org/fg5fkq">https://doi.org/fg5fkq</a><br />
DOI: <a href="https://doi.org/10.1063/1.1701475">10.1063/1.1701475</a></p>
</div>
<div id="ref-12mptWjWL">
<p>71. <strong>Corrected small basis set Hartree-Fock method for large systems</strong><br />
Rebecca Sure, Stefan Grimme<br />
<em>Journal of Computational Chemistry</em> (2013-07-15) <a href="https://doi.org/f42979">https://doi.org/f42979</a><br />
DOI: <a href="https://doi.org/10.1002/jcc.23317">10.1002/jcc.23317</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23670872">23670872</a></p>
</div>
</div>
<h2 class="page_break_before" id="SI">Supporting Information</h2>
<h3 id="the-mayer-bond-order-mbo-extends-the-wiberg-bond-order-wbo-to-non-minimal-basis-sets">The Mayer bond order (MBO) extends the Wiberg bond order (WBO) to non-minimal basis sets</h3>
<p>The WBO was developed within Pople’s complete neglect of differential overlap (CNDO) formalism <span class="citation" data-cites="19YdfyNRY BES2Ksiq">[<a href="#ref-BES2Ksiq" role="doc-biblioref">28</a>,<a href="#ref-19YdfyNRY" role="doc-biblioref">70</a>]</span> and is defined as:</p>
<p><span class="math display">\[\begin{equation}
    W_{AB} = \sum_{\mu \in A}\sum_{\nu \in B} |D_{\mu \nu}|^2
\end{equation}\]</span>{#eq:wiberg}</p>
<p>Where the quadradic sum of the electron density <span class="math inline">\(D\)</span> is over occupied orbitals <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\nu\)</span> in atoms A and B in the bond.
However, the WBO does not extend beyond semi-empirical methods where the overlap matrix is assumed to to be diagonal.
At higher levels of QM theory where non-minimal basis sets are used, <span class="math inline">\(W_{AB}\)</span> would give values that increase in size with the basis sets so they need to normalized.
In Psi4 <span class="citation" data-cites="YC6qkEYS">[<a href="#ref-YC6qkEYS" role="doc-biblioref">38</a>]</span>, The Löwdin normalization scheme <span class="citation" data-cites="K8bdDsCu">[<a href="#ref-K8bdDsCu" role="doc-biblioref">39</a>]</span> is used to calculate the WBOs.
In this scheme, the non-orthogonal basis set  are transformed by a linear transformation ’ = <span class="math inline">\(\bm{S}^{-\frac{1}{2}}\bm{\phi}\)</span> Where  is the the overlap matrix.</p>
<p>Mayer <span class="citation" data-cites="QlopodHU">[<a href="#ref-QlopodHU" role="doc-biblioref">35</a>]</span> incorporates  directly in the bond order calculation:</p>
<p><span class="math display">\[\begin{equation}
    B_{AB} = \sum_{\mu \in A}\sum_{\nu \in B} (\bm{DS})_{\mu\nu} (\bm{DS})_{\nu\mu}
\end{equation}\]</span>{#eq:mayer}</p>
<p>Where <span class="math inline">\(\bm{D}\)</span> is the density matrix for occupied orbitals for closed shell systems, and <span class="math inline">\(\bm{S}\)</span> is the overlap matrix.
This equation reduces to the WBO when orthonormalized basis sets are used.</p>
<h3 id="comparing-mbo-and-wbo-at-different-levels-of-qc-theory">Comparing MBO and WBO at different levels of QC theory</h3>
<p>Psi4 <span class="citation" data-cites="YC6qkEYS">[<a href="#ref-YC6qkEYS" role="doc-biblioref">38</a>]</span> implements both Wiberg-Löwdin and Mayer bond orders.
Here we compare WBO and MBOs at different levels of theory.</p>
<h4 id="differences-in-conformation-dependent-bond-order-distributions-between-wbo-and-mbo-at-different-levels-of-qc-theory">Differences in conformation-dependent bond order distributions between WBO and MBO at different levels of QC theory</h4>
<div id="fig:wbo-mbo-comparison" class="fignos">
<figure>
<img src="images/SI-wbo-dist.svg" alt="" /><figcaption><span>Figure 7:</span> <strong>Bond order distributions for rotatable bonds over 232 conformers for Mayer and Wiberg at different levels of QM theory.</strong>
<strong>[A]</strong> Gefitinb, with atoms numbered highlighted to correspond to numbers and colors on distribution plots.
<strong>[B]</strong> Distribution of AM1 WBOs for rotatable bonds.
<strong>[C]</strong> Same as <strong>[B]</strong> but for MBOs at HF3C.
<strong>[D]</strong> Same as <strong>[B]</strong> calculated at HF3C.
<strong>[E]</strong> Same as <strong>[B]</strong> but for MBOs at calculated at B3LYP.
<strong>[F]</strong> Same as <strong>[B]</strong> but calculated at B3LYP.</figcaption>
</figure>
</div>
<p>Both the WBO and MBO are functions of electron density which is conformation dependent, so they will change with conformation.
As we have shown in <a href="#fig:wbo_var_corr">2</a>, conjugated bonds have higher variance than non-conjugated bonds.
Different QC methods and ways of calculating bond orders result in differences in bond order values and conformation-dependent variance.
In Figure <a href="#fig:wbo-mbo-comparison">7</a> we compare AM1 <span class="citation" data-cites="tnSqySMX">[<a href="#ref-tnSqySMX" role="doc-biblioref">36</a>]</span> WBOs, HF-3c <span class="citation" data-cites="12mptWjWL">[<a href="#ref-12mptWjWL" role="doc-biblioref">71</a>]</span> MBOs and WBOs, and B3LYP(BJ)/DZVP <span class="citation" data-cites="YPDz9hB3 17YEERBX3 flKRoCNc CqesmIsy">[<a href="#ref-17YEERBX3" role="doc-biblioref">44</a>,<a href="#ref-flKRoCNc" role="doc-biblioref">45</a>,<a href="#ref-YPDz9hB3" role="doc-biblioref">46</a>,<a href="#ref-CqesmIsy" role="doc-biblioref">47</a>]</span> MBOs and WBOs.
Figure <a href="#fig:wbo-mbo-comparison">7</a> <strong>A</strong> shows Gefitinib, a kinase inhibitors, with its rotatable bonds highlighted to correspond to the colors of the distributions shown in the other panels of Figure <a href="#fig:wbo-mbo-comparison">7</a>.
The bond order distributions shown in Figure <a href="#fig:wbo-mbo-comparison">7</a> were calculated over 232 Omega <span class="citation" data-cites="1G1ccz54w">[<a href="#ref-1G1ccz54w" role="doc-biblioref">43</a>]</span> generated conformers of Gefitinib which were then geometry optimized with the methods used to calculate bond orders.</p>
<p>For AM1 we only calculated WBOs because the MBO and WBO are equivalent at that level of theory.
AM1 WBO distributions have lower variance than B3lyp for the non-conjugated bonds, but have significantly higher variance and more modes for the more conjugated bonds.</p>
<p>WBO and MBO distributions calculated from HF-3c optimized structures (Figure <a href="#fig:wbo-mbo-comparison">7</a> <strong>C</strong> and <strong>D</strong>) had lowest variances than other QC methods.
In addition, both MBO and WBOs have similar values and shapes of their distributions.
This might be a result of HF-3c using a small, Gaussian atomic orbital (AO) basis set.
As discussed in the previous section, the MBO reduces to the WBO when orthonormalized basis AO are used. In HF-3c, a small, Gaussian basis set is used which might be the reason why the MBOs and WBOs are so similar.</p>
<p>When using larger basis sets (B3LYP(BJ)/DZVP), the Mayer and Wiberg bond orders differ significantly.
First, MBOs have significantly higher variance with respect to conformation.
In addition, WBO values are significantly higher than Mayer and AM1 and HF3C WBOs.
The basis sets are Löwdin normalized, but the values still seem to grow with larger basis sets.</p>
<h4 id="differences-in-conformation-dependent-bond-order-correlation-structures-between-wbo-and-mbo-at-different-levels-of-qc-theory">Differences in conformation-dependent bond order correlation structures between WBO and MBO at different levels of QC theory</h4>
<div id="fig:correlations" class="fignos">
<figure>
<img src="images/SI-wbo-correlations.svg" alt="" /><figcaption><span>Figure 8:</span> <strong>Bond order correlations with respect to conformations for Mayer and Wiberg at different levels of QM theory.</strong>
<strong>[A]</strong> Gefitinb, with atoms numbered highlighted to correspond to numbers and colors on distribution plots.
<strong>[B]</strong> Pearson correlation coefficient of AM1 WBOs for every bond against every bond over 232 conformations.
<strong>[C]</strong> Same as <strong>[B]</strong> but for MBOs at HF3C.
<strong>[D]</strong> Same as <strong>[B]</strong> calculated at HF3C.
<strong>[E]</strong> Same as <strong>[B]</strong> but for MBOs at calculated at B3LYP.
<strong>[F]</strong> Same as <strong>[B]</strong> but calculated at B3LYP.</figcaption>
</figure>
</div>
<p>As we have shown in Figure <a href="#fig:wbo_var_corr">2</a>, conjugated systems have strongly correlated and anti-correlated bond orders with respect to conformations.
Figure <a href="#fig:correlations">8</a> compares such correlation maps and their resulting structures for the bond orders calculated in Figure <a href="#fig:wbo-mbo-comparison">7</a>.</p>
<p>Correlation plots from HF-3c MBOs and WBOs (Figures <a href="#fig:correlations">8</a> <strong>C</strong> and <strong>D</strong>) have structures that are very similar to each other.
This result is similar to Figures <a href="#fig:wbo-mbo-comparison">7</a> <strong>C</strong> and <strong>D</strong> where the distributions were also very similar.
However, the correlation structures of MBOs and WBOs calculated at B3LYP(BJ)/DZVP are different from each other.
MBOs show stronger correlations and anti-correlations for local conjugated systems.
For example, the bonds in the phenyl ring (alternating correlated and anti-correlated square formed by bonds 24-29) are all alternatively strongly correlated and anti-correlated.
In addition, bonds 19-23 and 23-24 are strongly correlated with each other and the quinazoline. The WBos calculated both at B3LYP(BJ)/DZVP and AM1 still show alternating correlated and anti-correlated bond orders for the phenyl ring, albeit not as strongly as seen for MBOs.
However, there is also a diffusion of the coupling to bonds outside of the phenyl ring. Bonds 24-25 and 24-29 are also coupled with quinazoline the bonds between the qunazoline and the phenyl (19-23 and 23-24).</p>
<p>From these results it is not clear which bond order is better at capturing electronic coupling in larger conjugated systems.
Further investigation is needed, specifically for deciding if MBO or WBO would be better at differentiating steric and resonance contributions to QC torsion scans discussed in .
It is encouraging to see that both AM1 and B3LYP(BJ)/DZVP WBOs generate such similar correlation structures.</p>
<h3 id="supplementary-figures">Supplementary Figures</h3>
<div id="fig:kinase_inhibitors" class="fignos">
<figure>
<img src="images/SI-KI-all.svg" alt="" /><figcaption><span>Figure 9:</span> <strong>Variance and correlations of WBO distributions with respect to conformations for a set of kinase inhibitors</strong>
This figure shows WBO distributions and correlations for a set of kinase inhibitors calculated at B3LYP-D3(BJ) / DZVP.
Optimized conformations and their WBOs are on QCArchive (<code>OptimizationDataset</code>, named <code>Kinase Inhibitors: WBO Distributions</code>)</figcaption>
</figure>
</div>
<div id="fig:si_wbo_dist" class="fignos">
<figure>
<img src="images/SI-gefitinib-wbo-dist.svg" alt="" /><figcaption><span>Figure 10:</span> <strong>Variance and correlations of AM1 Wiberg bond order distributions with respect to conformations are higher for conjugated bonds</strong>
The trend seen in <a href="#fig:wbo_var_corr">2</a> for higher level of QC theory (B3LYP-D3(BJ) / DZVP) is also observed at AM1.
<strong>[A]</strong> Gifitnib with its rotatable bonds highlighted. <strong>[B]</strong> WBO distributions over 232 conformations of the highlighted bonds. The variance
and their 95% CI is shown. <strong>[C]</strong> Pearson correlation coefficients for WBOs in every bond against all other bonds with respect to conformations.</figcaption>
</figure>
</div>
<div id="fig:wbo_dist_mols" class="fignos">
<figure>
<img src="images/druglike_mols_for_wbo_dist_4.svg" alt="" /><figcaption><span>Figure 11:</span> <strong>Druglike molecules used to calculate ELF10 AM1 WBOs</strong>
This set of molecules were selected to cover bonds of different types and multiplicities</figcaption>
</figure>
</div>
<div id="fig:hammett" class="fignos">
<figure>
<img src="images/SI-hammett.svg" alt="" /><figcaption><span>Figure 12:</span> <strong>Hammett parameters are anti-correlated with ELF10 WBOs</strong>
<strong>[A]</strong> Hammett sigma meta parameters vs AM1 ELF10 WBOs of X<sub>1</sub> meta to carboxylic acid.
<strong>[B]</strong> Same as <strong>A</strong> but for para substituents</figcaption>
</figure>
</div>
<div id="fig:phenyl_td_1" class="fignos">
<figure>
<img src="images/SI_substituted_phenyl_td_1.svg" alt="" /><figcaption><span>Figure 13:</span> <strong>QC torsion scans and corresponding WBOs for substituted phenyl set</strong>
QC torsion scans, WBOs corresponding to scan and torsion barrier heights vs ELF10 WBOs for phenyl set. QC scan colors correspond to highlighted central
bonds shown on the right. The molecules are labeled with their ELF10 WBOs.</figcaption>
</figure>
</div>
<div id="fig:phenyl_td_2" class="fignos">
<figure>
<img src="images/SI_substituted_phenyl_td_2.svg" alt="" /><figcaption><span>Figure 14:</span> <strong>QC torsion scans and corresponding WBOs for substituted phenyl set</strong>
QC torsion scans, WBOs corresponding to scan and torsion barrier heights vs ELF10 WBOs for phenyl set. QC scan colors correspond to highlighted central
bonds shown on the right. The molecules are labeled with their ELF10 WBOs.</figcaption>
</figure>
</div>
<div id="fig:phenyl_td_3" class="fignos">
<figure>
<img src="images/SI_substituted_phenyl_td_3.svg" alt="" /><figcaption><span>Figure 15:</span> <strong>QC torsion scans and corresponding WBOs for substituted phenyl set</strong>
QC torsion scans, WBOs corresponding to scan and torsion barrier heights vs ELF10 WBOs for phenyl set. QC scan colors correspond to highlighted central
bonds shown on the right. The molecules are labeled with their ELF10 WBOs.</figcaption>
</figure>
</div>
<div id="fig:phenyl_td_4" class="fignos">
<figure>
<img src="images/SI_substituted_phenyl_td_4.svg" alt="" /><figcaption><span>Figure 16:</span> <strong>QC torsion scans and corresponding WBOs for substituted phenyl set</strong>
QC torsion scans, WBOs corresponding to scan and torsion barrier heights vs ELF10 WBOs for phenyl set. QC scan colors correspond to highlighted central
bonds shown on the right. The molecules are labeled with their ELF10 WBOs.</figcaption>
</figure>
</div>
<div id="fig:phenyl_td_5" class="fignos">
<figure>
<img src="images/SI_substituted_phenyl_td_5.svg" alt="" /><figcaption><span>Figure 17:</span> <strong>QC torsion scans and corresponding WBOs for substituted phenyl set</strong>
QC torsion scans, WBOs corresponding to scan and torsion barrier heights vs ELF10 WBOs for phenyl set. QC scan colors correspond to highlighted central
bonds shown on the right. The molecules are labeled with their ELF10 WBOs.</figcaption>
</figure>
</div>
<section class="footnotes" role="doc-endnotes">
<hr />
<ol>
<li id="fn1" role="doc-endnote"><p>For non-minimal basis sets, AOs are often non-orthogonal and require normalization for the WBO to be valid. In the case of WBOs in Psi4 <span class="citation" data-cites="YC6qkEYS">[<a href="#ref-YC6qkEYS" role="doc-biblioref">38</a>]</span>, the Löwdin normalization <span class="citation" data-cites="K8bdDsCu 1BEyXJiMo">[<a href="#ref-K8bdDsCu" role="doc-biblioref">39</a>,<a href="#ref-1BEyXJiMo" role="doc-biblioref">40</a>]</span> scheme is used.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure img {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* heading 1 */
        h1 {
            font-size: 1.75em;
        }

        /* heading 2 */
        h2 {
            font-size: 1.25em;
            margin-top: 0;
        }

        /* heading 3 */
        h3 {
            font-size: 1.10em;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        .lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* side panel */
    .annotator-frame {
        width: 280px !important;
        z-index: 0 !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide toolbar and tooltip on print */
    @media only print {
        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- table of contents plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether default behavior is to be closed ('false'), open
            // ('true'), or only open when screen wide enough to fit panel
            // ('auto'). note: still always starts closed when page loads.
            open: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            document.body.insertBefore(panel, document.body.firstChild);

            closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // open/close panel based on option and screen size
        function openOrClosePanel() {
            if (
                options.open === 'true' ||
                (options.open === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                openOrClosePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector(
                '.icon_th_list'
            ).innerHTML;
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h3');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick() {
            openOrClosePanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- math plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
    });
</script>

<!-- math plugin -->

<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin configuration -->

<script>
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };
</script>

<!-- annotations plugin -->

<script src='https://hypothes.is/embed.js'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/
</script>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
